---
title: "Running EdgeR with per sample co variates on single cell data: R21 Sterile Immunity"
author: "Prasida Holla"
date: "2023-04-18"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Objective: 
1. Import CITE-seq ex vivo and multiome ex vivo Seurat objects with celltypes annotated based on reference mapping
  - CITE-seq: Samples are integrated as "apara" and "para" objects; split them out based on hashtag signal. Use AddMetaData to add a "Subject.ID" "Age" "Parasitemia" "Sex" columns to be used as co-variates downstream
  - Multiome: Isolate gene expression assays from each sample. Rename new.ident column to "Subject.ID" and add other co-variate columns
  
2. Integrate across all samples using integration anchors. Sanity check: See if old and new predicted celltype annotations match up (How?)

3. Generate uMAP plots at low and high resolution. Sanity check: Check cell type frequencies across samples from this integrated dataset matches with initial frequency assessments

4. DGE by MAST and/ or EdgeR using Sample.ID as a random effect

5. GSEA and check against old GSE results

# Load packages
```{r}
library(Seurat) 
library(SeuratWrappers)
library(patchwork)
library(SeuratDisk)
library(tidyverse)
library(googledrive)
library(ggpubr)
library(scales)
library(RColorBrewer)
library(viridis)
library(glue)
library(MAST)
```

# Import all datasets to be integrated
```{r import CITE seq and multiome ex vivo datasets, warning=T, results=T}
# CITE-seq ex vivo
# Apara
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1BlFyHZcotL7-myUYDmEit_FU8zyMN1zK"), path = temp, overwrite = TRUE)
apara_c <- readRDS(file = dl$local_path)
#Para
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1Bn-fCppOEOhTFxoWWNP46JUECSjcQk_e"), path = temp, overwrite = TRUE)
para_c <- readRDS(file = dl$local_path)

# Multiome
# Jun 2022
# Apara
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1_Id_-9KZj1lximQdYfsxnDpSYDwwLHiq"), path = temp, overwrite = TRUE)
apara_m <- readRDS(file = dl$local_path)
#Para
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1_M0Zo1iJKdfnr6_JDyhXN0MLlWLXHEDZ"), path = temp, overwrite = TRUE)
para_m <- readRDS(file = dl$local_path)

```

# Seprate out CITE-seq data into individual samples
```{r}
# Apara
rownames(apara_c[["Protein"]])
DefaultAssay(apara_c)<-"Protein"
FeatureScatter(apara_c, feature1 = "S1-Ex-vivo-PBMCs-kali0666-081621", feature2 = "S2-Ex-vivo-PBMCs-kali0618-081621", pt.size = 0.8)
s1c_k666<-subset(apara_c,subset =`S1-Ex-vivo-PBMCs-kali0666-081621`>2 & `S2-Ex-vivo-PBMCs-kali0618-081621`<2)
s2c_k618<-subset(apara_c,subset =`S1-Ex-vivo-PBMCs-kali0666-081621`<2 & `S2-Ex-vivo-PBMCs-kali0618-081621`>2)
s3c_k626<-subset(apara_c,subset =`S3-Ex-vivo-PBMCs-kali0626-081621`>2 & `S4-Ex-vivo-PBMCs-kali0636-081621`<2)
s4c_k636<-subset(apara_c,subset =`S3-Ex-vivo-PBMCs-kali0626-081621`<2 & `S4-Ex-vivo-PBMCs-kali0636-081621`>2)

#Para
rownames(para_c[["Protein"]])
DefaultAssay(para_c)<-"Protein"
s5c_k650<-subset(para_c,subset =`S5-Ex-vivo-PBMCs-kali0650-081621`>2 & `S6-Ex-vivo-PBMCs-kali0628-081621`<2)
s6c_k628<-subset(para_c,subset =`S5-Ex-vivo-PBMCs-kali0650-081621`<2 & `S6-Ex-vivo-PBMCs-kali0628-081621`>2)
s7c_k604<-subset(para_c,subset =`S7-Ex-vivo-PBMCs-kali0640-081621`>2 & `S8-Ex-vivo-PBMCs-kali0604-081621`<2)
s8c_k640<-subset(para_c,subset =`S7-Ex-vivo-PBMCs-kali0640-081621`<2 & `S8-Ex-vivo-PBMCs-kali0604-081621`>2)
```

# Seperate out multiome based on "new.ident"
```{r}
#Apara
apara_m.split<-SplitObject(apara_m, split.by = "new.ident")
k585<-apara_m.split$Kali0585_Apara_Batch1
k631<-apara_m.split$Kali0631_Apara_Batch1
#k613<-
k647<-apara_m.split$Kali0647_Apara_Batch2

#Para
para_m.split<-SplitObject(para_m, split.by = "new.ident")
k593<-para_m.split$Kali0585_Para_Batch1 # mislabeled in the original script "Multiome analysis: Differential gene expression (DGE) and GSEA analysis" but it is the correct sample. Will correct this
k600<-para_m.split$Kali0631_Para_Batch1 # mislabeled in the original script "Multiome analysis: Differential gene expression (DGE) and GSEA analysis" but it is the correct sample. Will correct this
k651<-para_m.split$Kali0651_Para_Batch2
k612<-para_m.split$Kali0612_Para_Batch2
```

#
# Add a column with unique sample IDs
```{r}
# CITE-seq
s1c_k666<-AddMetaData(s1c_k666,"Kali0666_Apara_Citeseq" , col.name = "Sample.ID")
s2c_k618<-AddMetaData(s2c_k618,"Kali0618_Apara_Citeseq" , col.name = "Sample.ID")
s3c_k626<-AddMetaData(s3c_k626,"Kali0626_Apara_Citeseq" , col.name = "Sample.ID")
s4c_k636<-AddMetaData(s4c_k636,"Kali0636_Apara_Citeseq" , col.name = "Sample.ID")
s5c_k650<-AddMetaData(s5c_k650,"Kali0650_Para_Citeseq" , col.name = "Sample.ID")
s6c_k628<-AddMetaData(s6c_k628,"Kali0628_Para_Citeseq" , col.name = "Sample.ID")
s7c_k604<-AddMetaData(s7c_k604,"Kali0604_Para_Citeseq" , col.name = "Sample.ID")
s8c_k640<-AddMetaData(s8c_k640,"Kali0640_Para_Citeseq" , col.name = "Sample.ID")
# Multiome
k585<-AddMetaData(k585,"Kali0585_Apara_Multiome" , col.name = "Sample.ID")
k631<-AddMetaData(k631,"Kali0631_Apara_Multiome" , col.name = "Sample.ID")
k593<-AddMetaData(k593,"Kali0585_Para_Multiome" , col.name = "Sample.ID")
k600<-AddMetaData(k600,"Kali0631_Para_Multiome" , col.name = "Sample.ID")
#k613<-AddMetaData(k585,"Kali0613_Apara_Multiome" , col.name = "Sample.ID") # not seqeunced as of 02-16-2023
k647<-AddMetaData(k647,"Kali0647_Apara_Multiome" , col.name = "Sample.ID")
k651<-AddMetaData(k651,"Kali0651_Para_Multiome" , col.name = "Sample.ID")
k612<-AddMetaData(k612,"Kali0612_Para_Multiome" , col.name = "Sample.ID")
```

# Add a batch column with aparasitemic and parasitemic annotations
```{r}
s1c_k666$batch = factor("Aparasitemic")
s2c_k618$batch = factor("Aparasitemic")
s3c_k626$batch = factor("Aparasitemic")
s4c_k636$batch = factor("Aparasitemic")
s5c_k650$batch = factor("Parasitemic")
s6c_k628$batch = factor("Parasitemic")
s7c_k604$batch = factor("Parasitemic")
s8c_k640$batch = factor("Parasitemic")
# Multiome
k585$batch = factor("Aparasitemic")
k631$batch = factor("Aparasitemic")
k593$batch = factor("Parasitemic")
k600$batch = factor("Parasitemic")
#k613$batch = factor("Aparasitemic")
k647$batch = factor("Aparasitemic")
k651$batch = factor("Parasitemic")
k612$batch = factor("Parasitemic")
```

# Equalize cell numbers across all samples by random sampling of larger ones
```{r}
## Evaluate the numer of cells in each
# CITE-seq
s1c_k666 #8209 cells
s2c_k618 #8254 cells
s3c_k626 #5847 cells
s4c_k636 #7829 cells
s5c_k650 #7275 cells
s6c_k628 #8179 cells
s7c_k604 #6679 cells
s8c_k640 #8143 cells

#Multiome
k585 #6803 cells
k631 #8448 cells
k593 #9652 cells
k600 #8853 cells
#k613
k647 #4834 cells
k651 #1431 cells
k612 #6424 cells


# sample with least #K651 of cells: , which has 1431 cells. We will reduce all others to 1431 cells (will make computation of the integrated object easier but is a massive loss of cells)
set.seed() # check with different seeds
s1c_k666<- s1c_k666[, sample(colnames(s1c_k666), size = ncol(k651), replace=F)]
s2c_k618<- s2c_k618[, sample(colnames(s2c_k618), size = ncol(k651), replace=F)]
s3c_k626<- s3c_k626[, sample(colnames(s3c_k626), size = ncol(k651), replace=F)]
s4c_k636<- s4c_k636[, sample(colnames(s4c_k636), size = ncol(k651), replace=F)]
s5c_k650<- s5c_k650[, sample(colnames(s5c_k650), size = ncol(k651), replace=F)]
s6c_k628<- s6c_k628[, sample(colnames(s6c_k628), size = ncol(k651), replace=F)]
s7c_k604<- s7c_k604[, sample(colnames(s7c_k604), size = ncol(k651), replace=F)]
s8c_k640<- s8c_k640[, sample(colnames(s8c_k640), size = ncol(k651), replace=F)]
k585<- k585[, sample(colnames(k585), size = ncol(k651), replace=F)]
k631<- k631[, sample(colnames(k631), size = ncol(k651), replace=F)]
k593<- k593[, sample(colnames(k593), size = ncol(k651), replace=F)]
k600<- k600[, sample(colnames(k600), size = ncol(k651), replace=F)]
k647<- k647[, sample(colnames(k647), size = ncol(k651), replace=F)]
k612<- k612[, sample(colnames(k612), size = ncol(k651), replace=F)]
```

# For this purpose, we only want RNA assays
```{r}
DefaultAssay(s1c_k666)<-"RNA"
DefaultAssay(s2c_k618)<-"RNA"
DefaultAssay(s3c_k626)<-"RNA"
DefaultAssay(s4c_k636)<-"RNA"
DefaultAssay(s5c_k650)<-"RNA"
DefaultAssay(s6c_k628)<-"RNA"
DefaultAssay(s7c_k604)<-"RNA"
DefaultAssay(s8c_k640)<-"RNA"
DefaultAssay(k585)<-"RNA"
DefaultAssay(k631)<-"RNA"
DefaultAssay(k593)<-"RNA"
DefaultAssay(k600)<-"RNA"
#DefaultAssay(k613)<-"RNA"
DefaultAssay(k647)<-"RNA"
DefaultAssay(k651)<-"RNA"
DefaultAssay(k612)<-"RNA"
```

# Integrate all 15 samples (except Kali0613, which has not been seqeunced yet)
```{r}
# Step 1: Find integration anchors across each of the objects and integrate
all<-merge(s1c_k666, c(s2c_k618,
            s3c_k626,
           s4c_k636,
           s5c_k650,
           s6c_k628,
           s7c_k604,
           s8c_k640,
           k585,
           k631,
           k593,
           k600,
           k647,
           k651,
           k612),
           add.cell.ids = c("Kali0666_Apara_Citeseq",
                            "Kali0618_Apara_Citeseq",
                            "Kali0626_Apara_Citeseq",
                            "Kali0636_Apara_Citeseq",
                            "Kali0650_Para_Citeseq",
                            "Kali0628_Para_Citeseq",
                            "Kali0604_Para_Citeseq",
                            "Kali0640_Para_Citeseq",
                            "Kali0585_Apara_Multiome",
                            "Kali0631_Apara_Multiome", 
                            "Kali0585_Para_Multiome",   #### FIX THIS!!!!
                            "Kali0631_Para_Multiome",     #### FIX THIS!!!!
                            "Kali0647_Apara_Multiome",
                            "Kali0651_Para_Multiome",
                            "Kali0612_Para_Multiome"))

all.list <- SplitObject(all, split.by = "Sample.ID")  

for (i in 1:length(all.list)) {
  all.list[[i]] <- NormalizeData(all.list[[i]], verbose = FALSE)
  all.list[[i]] <- FindVariableFeatures(all.list[[i]], selection.method = "vst",
                                           nfeatures = 2000,verbose = FALSE)
}

all.anchors<-FindIntegrationAnchors(all.list, dim=1:50,anchor.features = 2000,
                                       l2.norm = T, k.anchor = 5,k.score = 20,
                                       max.features = 200)

all.integrated <- IntegrateData(anchorset = all.anchors, dims = 1:50)

# Step 2: Reduce and run dimensionality reduction

DefaultAssay(all.integrated) <- "integrated"
all.integrated <- ScaleData(all.integrated, verbose = FALSE)
all.integrated <- RunPCA(all.integrated, npcs = 50, verbose = FALSE)

# Check optimal number of PCs/ dimensions to be included for dim.reduction. There are several ways to do this
# 1. Identify the top 10-50 most highly variable genes. This approach is more useful for a relatively pure celltype than a mixture of celltypes such as PBMC
top50 <- head(VariableFeatures(all.integrated), 50)

# 2. Elbow plot of PCs vs variance
ElbowPlot(all.integrated) # shows that beyond 6 PCs, variance drops below 2 and this can be a cutoff

# 3. Heatmap of PCs to see where variance is breaking down (looks to be ~ 6PCs)
#DimHeatmap(all.integrated, dims = 1:20, cells = 500, balanced = TRUE)

# Based on these, we will use 6 PCs for uMAP and tSNE analysis
all.integrated <- RunUMAP(all.integrated, reduction = "pca", dims = 1:15)
all.integrated <- RunTSNE(all.integrated, reduction = "pca", dims = 1:15, check_duplicates=F)

all.integrated <- FindNeighbors(all.integrated, reduction = "pca", 
                                   dims = 1:15)
all.integrated <- FindClusters(all.integrated, resolution = 0.5)

#Step 3a: Load the reference dataset (note: this is hard to upload from google Drive dur to its large size)
reference <- LoadH5Seurat("~/Desktop/R21 Sterile Immunity data and scripts/CITE-seq/pbmc_multimodal.h5seurat")

# Step 3b: SCTransform the query datasets
all_sc<- SCTransform(all.integrated, verbose = FALSE) 

# Step 3c: Find anchors between reference and query datasets
all.anchors<- FindTransferAnchors(
  reference = reference,
  query = all_sc, 
  normalization.method = "SCT",
  reference.reduction = "spca",
  dims = 1:50
)

# Step 3d: Run MapQuery, which transfers cell type labels and protein data from reference to query
#5y
refmapped.all <- MapQuery(
  anchorset = all.anchors,
  query = all_sc,
  reference = reference,
  refdata = list(
    celltype.l1 = "celltype.l1",
    celltype.l2 = "celltype.l2",
    predicted_ADT = "ADT"
  ),
  reference.reduction = "spca", 
  reduction.model = "wnn.umap"
)

saveRDS(refmapped.all, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/Refmapped_and_dim_reduced_all15_minusK613_04_24_2023.rds")

# Import here from Google Drive
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("10EKJ-eE2_oouFiHTE5vwMfLF6qVZj_kc"), path = temp, overwrite = TRUE)
refmapped.all<- readRDS(file = dl$local_path)
```
# Separate by batch
```{r}
all.split<-SplitObject(refmapped.all, split.by = "batch")
all_para<-all.split$Parasitemic
all_apara<-all.split$Aparasitemic
```


Seperate out individual ceclltypes
```{r}
#Para
# Low resolution celltypes
list_para_l1 <- c()
for (i in levels(factor(all_para[["predicted.celltype.l1"]][[1]]))){
  list_para_l1[[i]] <- subset(all_para, subset=`predicted.celltype.l1`==i)
}

# High resolution celltypes
list_para_l2 <- c()
for (i in levels(factor(all_para[["predicted.celltype.l2"]][[1]]))){
  list_para_l2[[i]] <- subset(all_para, subset=`predicted.celltype.l2`==i)
}

#Apara
# Low resolution celltypes
list_apara_l1 <- c()
for (i in levels(factor(all_apara[["predicted.celltype.l1"]][[1]]))){
  list_apara_l1[[i]] <- subset(all_apara, subset=`predicted.celltype.l1`==i)
}

# High resolution celltypes
list_apara_l2 <- c()
for (i in levels(factor(all_apara[["predicted.celltype.l2"]][[1]]))){
  list_apara_l2[[i]] <- subset(all_apara, subset=`predicted.celltype.l2`==i)
}
```

#split out individual celltypes
```{r}
# Predicted celltype L1

#apara
# Check indices prior to assignment for seperation of cell types
names(list_apara_l1)
b_apar_l1<-list_apara_l1[[1]]
cd4t_apar_l1<-list_apara_l1[[2]]
cd8t_apar_l1<-list_apara_l1[[3]]
dc_apar_l1<-list_apara_l1[[4]]
mono_apar_l1<-list_apara_l1[[5]]
nk_apar_l1<-list_apara_l1[[6]]
otherT_apar_l1<-list_apara_l1[[8]]
other_apar_l1<-list_apara_l1[[7]]
#para
names(list_apara_l1)
b_par_l1<-list_para_l1[[1]]
cd4t_par_l1<-list_para_l1[[2]]
cd8t_par_l1<-list_para_l1[[3]]
dc_par_l1<-list_para_l1[[4]]
mono_par_l1<-list_para_l1[[5]]
nk_par_l1<-list_para_l1[[6]]
otherT_par_l1<-list_para_l1[[8]]
other_par_l1<-list_para_l1[[7]]

# predicted celltype L2
#apara
names(list_apara_l2)
#asdc_apar_l2<-list_apara_l2[[]] # no ASDCs in this object
b_int_apar_l2<-list_apara_l2[[1]]
b_mem_apar_l2<-list_apara_l2[[2]]
b_naiv_apar_l2<-list_apara_l2[[3]]
pb_apar_l2<-list_apara_l2[[25]]
cd14_mono_apar_l2<-list_apara_l2[[4]]
cd16_mono_apar_l2<-list_apara_l2[[5]]
cd4_ctl_apar_l2<-list_apara_l2[[6]]
cd4_naiv_apar_l2<-list_apara_l2[[7]]
cd4_prolif_apar_l2<-list_apara_l2[[8]]
cd4_tcm_apar_l2<-list_apara_l2[[9]]
cd4_tem_apar_l2<-list_apara_l2[[10]]
cd8_naiv_apar_l2<-list_apara_l2[[11]]
cd8_tcm_apar_l2<-list_apara_l2[[12]]
cd8_tem_apar_l2<-list_apara_l2[[13]]
treg_apar_l2<-list_apara_l2[[27]]
gdt_apar_l2<-list_apara_l2[[17]]
mait_apar_l2<-list_apara_l2[[20]]
ilc_apar_l2<-list_apara_l2[[19]]
pdc_apar_l2<-list_apara_l2[[24]]
cdc2_apar_l2<-list_apara_l2[[14]]
nk_apar_l2<-list_apara_l2[[21]]
nk_cd56br_apar_l2<-list_apara_l2[[23]]
hspc_apar_l2<-list_apara_l2[[18]]
dnt_apar_l2<-list_apara_l2[[15]]
nkprolif_apar_l2<-list_apara_l2[[22]]
platelet_apar_l2<-list_apara_l2[[26]]
#para
names(list_para_l2)
asdc_par_l2<-list_para_l2[[1]]
b_int_par_l2<-list_para_l2[[2]]
b_mem_par_l2<-list_para_l2[[3]]
b_naiv_par_l2<-list_para_l2[[4]]
pb_par_l2<-list_para_l2[[27]]
cd14_mono_par_l2<-list_para_l2[[5]]
cd16_mono_par_l2<-list_para_l2[[6]]
cd4_ctl_par_l2<-list_para_l2[[7]]
cd4_naiv_par_l2<-list_para_l2[[8]]
cd4_prolif_par_l2<-list_para_l2[[9]]
cd4_tcm_par_l2<-list_para_l2[[10]]
cd4_tem_par_l2<-list_para_l2[[11]]
cd8_naiv_par_l2<-list_para_l2[[12]]
#cd8_prolif_par_l2<-list_para_l2[[13]]
cd8_tcm_par_l2<-list_para_l2[[14]]
cd8_tem_par_l2<-list_para_l2[[15]]
treg_par_l2<-list_para_l2[[29]]
gdt_par_l2<-list_para_l2[[19]]
mait_par_l2<-list_para_l2[[22]]
ilc_par_l2<-list_para_l2[[21]]
pdc_par_l2<-list_para_l2[[26]]
cdc2_par_l2<-list_para_l2[[16]]
nk_par_l2<-list_para_l2[[23]]
nk_cd56br_par_l2<-list_para_l2[[25]]
hspc_par_l2<-list_para_l2[[20]]
dnt_par_l2<-list_para_l2[[17]]
nkprolif_par_l2<-list_para_l2[[24]]
platelet_par_l2<-list_para_l2[[28]]
```

#Merge parasitemic and aparasitemic objects for each celltype
```{r}
# predicted celltype l1
# Apara
Idents(b_apar_l1)<-"apara"
Idents(cd4t_apar_l1)<-"apara"
Idents(cd8t_apar_l1)<-"apara"
Idents(dc_apar_l1)<-"apara"
Idents(mono_apar_l1)<-"apara"
Idents(nk_apar_l1)<-"apara"
Idents(otherT_apar_l1)<-"apara"
Idents(other_apar_l1)<-"apara"
# Para
Idents(b_par_l1)<-"para"
Idents(cd4t_par_l1)<-"para"
Idents(cd8t_par_l1)<-"para"
Idents(dc_par_l1)<-"para"
Idents(mono_par_l1)<-"para"
Idents(nk_par_l1)<-"para"
Idents(otherT_par_l1)<-"para"
Idents(other_par_l1)<-"para"

# predicted celltype l2
# Apara
#Idents(asdc_apar_l2)<-"apara"
Idents(b_int_apar_l2)<-"apara"
Idents(b_mem_apar_l2)<-"apara"
Idents(b_naiv_apar_l2)<-"apara"
Idents(pb_apar_l2)<-"apara"
Idents(cd14_mono_apar_l2)<-"apara"
Idents(cd16_mono_apar_l2)<-"apara"
Idents(cd4_ctl_apar_l2)<-"apara"
Idents(cd4_naiv_apar_l2)<-"apara"
Idents(cd4_prolif_apar_l2)<-"apara"
Idents(cd4_tcm_apar_l2)<-"apara"
Idents(cd4_tem_apar_l2)<-"apara"
Idents(cd8_naiv_apar_l2)<-"apara"
Idents(cd8_tcm_apar_l2)<-"apara"
Idents(cd8_tem_apar_l2)<-"apara"
Idents(treg_apar_l2)<-"apara"
Idents(gdt_apar_l2)<-"apara"
Idents(mait_apar_l2)<-"apara"
Idents(ilc_apar_l2)<-"apara"
Idents(pdc_apar_l2)<-"apara"
Idents(cdc2_apar_l2)<-"apara"
Idents(nk_apar_l2)<-"apara"
Idents(nk_cd56br_apar_l2)<-"apara"
Idents(hspc_apar_l2)<-"apara"
Idents(dnt_apar_l2)<-"apara"
Idents(nkprolif_apar_l2)<-"apara"
Idents(platelet_apar_l2)<-"apara"
# Para
Idents(asdc_par_l2)<-"para"
Idents(b_int_par_l2)<-"para"
Idents(b_mem_par_l2)<-"para"
Idents(b_naiv_par_l2)<-"para"
Idents(pb_par_l2)<-"para"
Idents(cd14_mono_par_l2)<-"para"
Idents(cd16_mono_par_l2)<-"para"
Idents(cd4_ctl_par_l2)<-"para"
Idents(cd4_naiv_par_l2)<-"para"
Idents(cd4_prolif_par_l2)<-"para"
Idents(cd4_tcm_par_l2)<-"para"
Idents(cd4_tem_par_l2)<-"para"
Idents(cd8_naiv_par_l2)<-"para"
Idents(cd8_tcm_par_l2)<-"para"
Idents(cd8_tem_par_l2)<-"para"
Idents(treg_par_l2)<-"para"
Idents(gdt_par_l2)<-"para"
Idents(mait_par_l2)<-"para"
Idents(ilc_par_l2)<-"para"
Idents(pdc_par_l2)<-"para"
Idents(cdc2_par_l2)<-"para" 
Idents(nk_par_l2)<-"para"
Idents(nk_cd56br_par_l2)<-"para"
Idents(hspc_par_l2)<-"para"
Idents(dnt_par_l2)<-"para"
Idents(nkprolif_par_l2)<-"para"
Idents(platelet_par_l2)<-"para"

# Merge the aparasitemic and parasitemic objects together
# Predicted celltype l1
bcell_merged_l1<-merge(b_par_l1, b_apar_l1, add.cell.ids=c("para", "apara"))
cd4t_merged_l1<-merge(cd4t_par_l1, cd4t_apar_l1, add.cell.ids=c("para", "apara"))
cd8t_merged_l1<-merge(cd8t_par_l1, cd8t_apar_l1, add.cell.ids=c("para", "apara"))
dc_merged_l1<-merge(dc_par_l1, dc_apar_l1, add.cell.ids=c("para", "apara"))
mono_merged_l1<-merge(mono_par_l1, mono_apar_l1, add.cell.ids=c("para", "apara"))
nk_merged_l1<-merge(nk_par_l1, nk_apar_l1, add.cell.ids=c("para", "apara"))
otherT_merged_l1<-merge(otherT_par_l1, otherT_apar_l1, add.cell.ids=c("para", "apara"))
other_merged_l1<-merge(other_par_l1, other_apar_l1, add.cell.ids=c("para", "apara"))

# Predicted celltype l2
#asdc_merged_l2<-merge(asdc_par_l2, asdc_apar_l2, add.cell.ids=c("para", "apara"))
b_int_merged_l2<-merge(b_int_par_l2, b_int_apar_l2, add.cell.ids=c("para", "apara"))
b_mem_merged_l2<-merge(b_mem_par_l2, b_mem_apar_l2, add.cell.ids=c("para", "apara"))
b_naiv_merged_l2<-merge(b_naiv_par_l2, b_naiv_apar_l2, add.cell.ids=c("para", "apara"))
pb_merged_l2<-merge(pb_par_l2, pb_apar_l2, add.cell.ids=c("para", "apara"))
cd14_mono_merged_l2<-merge(cd14_mono_par_l2, cd14_mono_apar_l2, add.cell.ids=c("para", "apara"))
cd16_mono_merged_l2<-merge(cd16_mono_par_l2, cd16_mono_apar_l2, add.cell.ids=c("para", "apara"))
cd4_ctl_merged_l2<-merge(cd4_ctl_par_l2, cd4_ctl_apar_l2, add.cell.ids=c("para", "apara"))
cd4_naiv_merged_l2<-merge(cd4_naiv_par_l2, cd4_naiv_apar_l2, add.cell.ids=c("para", "apara"))
cd4_prolif_merged_l2<-merge(cd4_prolif_par_l2, cd4_prolif_apar_l2, add.cell.ids=c("para", "apara"))
cd4_tcm_merged_l2<-merge(cd4_tcm_par_l2, cd4_tcm_apar_l2, add.cell.ids=c("para", "apara"))
cd4_tem_merged_l2<-merge(cd4_tem_par_l2, cd4_tem_apar_l2, add.cell.ids=c("para", "apara"))
cd8_naiv_merged_l2<-merge(cd8_naiv_par_l2, cd8_naiv_apar_l2, add.cell.ids=c("para", "apara"))
cd8_tcm_merged_l2<-merge(cd8_tcm_par_l2, cd8_tcm_apar_l2, add.cell.ids=c("para", "apara"))
cd8_tem_merged_l2<-merge(cd8_tem_par_l2, cd8_tem_apar_l2, add.cell.ids=c("para", "apara"))
treg_merged_l2<-merge(treg_par_l2, treg_apar_l2, add.cell.ids=c("para", "apara"))
gdt_merged_l2<-merge(gdt_par_l2, gdt_apar_l2, add.cell.ids=c("para", "apara"))
mait_merged_l2<-merge(mait_par_l2, mait_apar_l2, add.cell.ids=c("para", "apara"))
ilc_merged_l2<-merge(ilc_par_l2, ilc_apar_l2, add.cell.ids=c("para", "apara"))
pdc_merged_l2<-merge(pdc_par_l2, pdc_apar_l2, add.cell.ids=c("para", "apara"))
cdc2_merged_l2<-merge(cdc2_par_l2, cdc2_apar_l2, add.cell.ids=c("para", "apara")) # throws error, possibly due to single cell in cdc2_apar_l2
nk_merged_l2<-merge(nk_par_l2, nk_apar_l2, add.cell.ids=c("para", "apara"))
nk_cd56br_merged_l2<-merge(nk_cd56br_par_l2, nk_cd56br_apar_l2, add.cell.ids=c("para", "apara"))
hspc_merged_l2<-merge(hspc_par_l2, hspc_apar_l2, add.cell.ids=c("para", "apara"))
dnt_merged_l2<-merge(dnt_par_l2, dnt_apar_l2, add.cell.ids=c("para", "apara"))
nkprolif_merged_l2<-merge(nkprolif_par_l2, nkprolif_apar_l2, add.cell.ids=c("para", "apara"))
platelet_merged_l2<-merge(platelet_par_l2, platelet_apar_l2, add.cell.ids=c("para", "apara"))# throws error, possibly due to single cell in platelet_par_l2
```


```{r}
# To make the merged objects comparable, run PrepSCTFindMarkers on each object
# Predicted celltype l1
bcell_merged_l1<-PrepSCTFindMarkers(bcell_merged_l1)
cd4t_merged_l1<-PrepSCTFindMarkers(cd4t_merged_l1)
cd8t_merged_l1<-PrepSCTFindMarkers(cd8t_merged_l1)
dc_merged_l1<-PrepSCTFindMarkers(dc_merged_l1)
mono_merged_l1<-PrepSCTFindMarkers(mono_merged_l1)
nk_merged_l1<-PrepSCTFindMarkers(nk_merged_l1)
otherT_merged_l1<-PrepSCTFindMarkers(otherT_merged_l1)
other_merged_l1<-PrepSCTFindMarkers(other_merged_l1)

# Predicted celltype l2
#asdc_merged_l2<-PrepSCTFindMarkers(asdc_merged_l2)
b_int_merged_l2<-PrepSCTFindMarkers(b_int_merged_l2)
b_mem_merged_l2<-PrepSCTFindMarkers(b_mem_merged_l2)
b_naiv_merged_l2<-PrepSCTFindMarkers(b_naiv_merged_l2)
pb_merged_l2<-PrepSCTFindMarkers(pb_merged_l2)
cd14_mono_merged_l2<-PrepSCTFindMarkers(cd14_mono_merged_l2)
cd16_mono_merged_l2<-PrepSCTFindMarkers(cd16_mono_merged_l2)
cd4_ctl_merged_l2<-PrepSCTFindMarkers(cd4_ctl_merged_l2)
cd4_naiv_merged_l2<-PrepSCTFindMarkers(cd4_naiv_merged_l2)
cd4_prolif_merged_l2<-PrepSCTFindMarkers(cd4_prolif_merged_l2)
cd4_tcm_merged_l2<-PrepSCTFindMarkers(cd4_tcm_merged_l2)
cd4_tem_merged_l2<-PrepSCTFindMarkers(cd4_tem_merged_l2)
cd8_naiv_merged_l2<-PrepSCTFindMarkers(cd8_naiv_merged_l2)
cd8_tcm_merged_l2<-PrepSCTFindMarkers(cd8_tcm_merged_l2)
cd8_tem_merged_l2<-PrepSCTFindMarkers(cd8_tem_merged_l2)
treg_merged_l2<-PrepSCTFindMarkers(treg_merged_l2)
gdt_merged_l2<-PrepSCTFindMarkers(gdt_merged_l2)
mait_merged_l2<-PrepSCTFindMarkers(mait_merged_l2)
ilc_merged_l2<-PrepSCTFindMarkers(ilc_merged_l2)
pdc_merged_l2<-PrepSCTFindMarkers(pdc_merged_l2)
cdc2_merged_l2<-PrepSCTFindMarkers(cdc2_merged_l2)
nk_merged_l2<-PrepSCTFindMarkers(nk_merged_l2)
nk_cd56br_merged_l2<-PrepSCTFindMarkers(nk_cd56br_merged_l2)
hspc_merged_l2<-PrepSCTFindMarkers(hspc_merged_l2)
dnt_merged_l2<-PrepSCTFindMarkers(dnt_merged_l2)
nkprolif_merged_l2<-PrepSCTFindMarkers(nkprolif_merged_l2)
platelet_merged_l2<-PrepSCTFindMarkers(platelet_merged_l2)

```

```{r}
bcell_deglist_l1<- FindMarkers(bcell_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd4t_deglist_l1<- FindMarkers(cd4t_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd8t_deglist_l1<- FindMarkers(cd8t_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
dc_deglist_l1<- FindMarkers(dc_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
mono_deglist_l1<- FindMarkers(mono_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
nk_deglist_l1<- FindMarkers(nk_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
otherT_deglist_l1<- FindMarkers(otherT_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
other_deglist_l1<- FindMarkers(other_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)


# Predicted celltype L2
#asdc_deglist_l2 <- FindMarkers(asdc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
b_int_deglist_l2 <- FindMarkers(b_int_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
b_mem_deglist_l2 <- FindMarkers(b_mem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
b_naiv_deglist_l2 <- FindMarkers(b_naiv_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
pb_deglist_l2 <- FindMarkers(pb_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd14_mono_deglist_l2 <- FindMarkers(cd14_mono_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd16_mono_deglist_l2 <- FindMarkers(cd16_mono_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd4_ctl_deglist_l2 <- FindMarkers(cd4_ctl_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd4_naiv_deglist_l2 <- FindMarkers(cd4_naiv_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd4_prolif_deglist_l2 <- FindMarkers(cd4_prolif_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd4_tcm_deglist_l2 <- FindMarkers(cd4_tcm_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd4_tem_deglist_l2 <- FindMarkers(cd4_tem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd8_naiv_deglist_l2 <- FindMarkers(cd8_naiv_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd8_tcm_deglist_l2 <- FindMarkers(cd8_tcm_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cd8_tem_deglist_l2 <- FindMarkers(cd8_tem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
treg_deglist_l2 <- FindMarkers(cd8_tem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
gdt_deglist_l2 <- FindMarkers(gdt_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
mait_deglist_l2 <- FindMarkers(mait_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
ilc_deglist_l2 <- FindMarkers(ilc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
pdc_deglist_l2 <- FindMarkers(pdc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
cdc2_deglist_l2 <- FindMarkers(cdc2_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
nk_deglist_l2 <- FindMarkers(nk_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
nk_cd56br_deglist_l2 <- FindMarkers(nk_cd56br_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
hspc_deglist_l2 <- FindMarkers(hspc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
dnt_deglist_l2 <- FindMarkers(dnt_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
nkprolif_deglist_l2 <- FindMarkers(nkprolif_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, min.pct = 0.1)
```

# Concatenate all dataframes of DEGlists together
```{r}
allcelltypes_exvivo_deglists <- mget(ls(pattern = "*deglist"))

# Save
saveRDS(allcelltypes_exvivo_deglists, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/R21SterileImmunity_All15_samples_CITE_Multiome_allcelltypes_DEGList_Parasitemic_vs_Aparasitemic_test_MAST_with_latent.vars_SampleID_04_25_2023.rds")
```

# Alternately, use these custom function modification to FindMarkers and MAST's MASTDETest, which are specifically tailored for adding random effect. Check these results against original DE lists, using only latent.vars in FindMarkers
```{r}
# First set of modifications are MAST helper functions. They come from https://github.com/RGLab/MAST/blob/devel/R/helper-methods.R Importantly, this allows you to incorporate the function new_with_repaired_slots, which will be used in the next step
##' Compute the Et from the Ct
##'
##' Computes the Et value from the Ct value in an existing data frame and returns a new data frame with the Et column appended
##' @param df a \code{data.frame}
##' @param column The name of the \code{Ct} column. A \code{character}. 'Ct' by default.
##' @param Cmax the maximum number of cycles performed. 40 by default.
##' @return A copy of \code{df} with the 'Et' column appended
##' @author Greg Finak
##' @export
##' @examples
##' data(vbeta)
##' vbeta <- computeEtFromCt(vbeta)
computeEtFromCt<-function(df,column='Ct',Cmax=40){
    within.data.frame(df, {Et <- Cmax-get(column); Et <- ifelse(is.na(Et), 0, Et)})
}

reraise <- function(err, convertToWarning=FALSE, silent=FALSE){
    if(silent) return(err)
    if(convertToWarning){
        warning(simpleWarning(message=err$message, call=err$call))
    } else{
        stop(simpleError(message=err$message, call=err$call))
    }
    return(err)
}

##' Selectively muffle warnings based on output
##'
##' @param expr an expression
##' @param regexp a regexp to be matched (with str_detect)
##' @return the result of expr
##' @examples
##' hushWarning(warning('Beware the rabbit'), 'rabbit')
##' hushWarning(warning('Beware the rabbit'), 'hedgehog')
##' @export
hushWarning <- function(expr, regexp){
    withCallingHandlers(expr, warning=function(w){
        if(str_detect(conditionMessage(w), regexp)) invokeRestart("muffleWarning")
    })
}

##' Remove the left hand side (response) from a formula
##'
##' The order of terms will be rearrange to suit R's liking for hierarchy but otherwise the function should be idempotent for
##' @param Formula formula
##' @param warn Issue a warning if a response variable is found?
##' @return formula
##' @author Andrew
removeResponse <- function(Formula, warn=TRUE){
    charForm <- paste0(deparse(Formula, width.cutoff=500), collapse='')
    fsplit <- str_split_fixed(charForm, fixed('~'), 2)
    if(nchar(fsplit[1,1])>0 && warn) message("Ignoring LHS of formula (", fsplit[1,1], ') and using assay(sca)')
    as.formula(paste0('~', fsplit[1,2])) 
}

#' Combine lists, preferentially taking elements from x if there are duplicate names
#'
#' @param x list
#' @param y list
#' @examples
#' MAST:::meld_list_left(list(A=1, B=2), list(A = 0))
meld_list_left = function(x, y){
    unite = c(x, y)
    dups = nchar(names(unite)) & duplicated(names(unite))
    unite[!dups]
}

#' Instantiate a class, but warn rather than error for badly named slots
#'
#' @param classname `character` naming a class
#' @param ... slots in `classname`
#' @param extra named list giving other slots in `classname`
#'
#' @return `new(classname)`
#' @examples
#' MAST:::new_with_repaired_slots("SimpleList",  listData = list(x = LETTERS), 
#' extra = list(elementType = 'character', food = "tasty", beer = "cold"))
new_with_repaired_slots = function(classname, ..., extra){
  #... were actually named in caller, so assumed good.
  # extra are dot args, and might have been passed in erroneously by a caller.
  bad_slots = setdiff(names(extra), slotNames(classname))
  if(length(bad_slots) > 0){
    warning(sprintf("Dropping illegal slot(s) %s for class %s.  
                    This likely indicates a bug in an upstream package.", 
                    paste0(bad_slots, collapse = ', '), classname))
  }
  do.call(new, c(Class = classname, list(...), extra[setdiff(names(extra), bad_slots)]))
}

# Second set of functions is from https://github.com/satijalab/seurat/issues/3712, which allows modification of MAST and Seurat funcitons

# We want a different behavior from MASTDETest, where the latent variable is
# used as a random effect in the model
rem_MASTDETest <- function(
  data.use,
  cells.1,
  cells.2,
  latent.vars = NULL,
  verbose = TRUE,
  # New option - random effect variable (should be included in latent.vars)
  re.var = NULL,
  ...
) {
  # Check for MAST
  if (!PackageCheck('MAST', error = FALSE)) {
    stop("Please install MAST - learn more at https://github.com/RGLab/MAST")
  }
  group.info <- data.frame(row.names = c(cells.1, cells.2))
  latent.vars <- latent.vars %||% group.info
  group.info[cells.1, "group"] <- "Group1"
  group.info[cells.2, "group"] <- "Group2"
  group.info[, "group"] <- factor(x = group.info[, "group"])
  latent.vars.names <- c("condition", colnames(x = latent.vars))
  latent.vars <- cbind(latent.vars, group.info)
  latent.vars$wellKey <- rownames(x = latent.vars)
  fdat <- data.frame(rownames(x = data.use))
  colnames(x = fdat)[1] <- "primerid"
  rownames(x = fdat) <- fdat[, 1]
  sca <- MAST::FromMatrix(
    exprsArray = as.matrix(x = data.use),
    check_sanity = FALSE,
    cData = latent.vars,
    fData = fdat
  )
  cond <- factor(x = SummarizedExperiment::colData(sca)$group)
  cond <- relevel(x = cond, ref = "Group1")
  SummarizedExperiment::colData(sca)$condition <- cond

  # This is the main change in the code - we want ~ ( 1 | re.var) in the formula:
  # ~ condition + lat.vars + (1 | re.var)
  if (!is.null(re.var)) {
    if (!re.var %in% latent.vars.names) {
      stop("Random effect variable should be included in latent variables!")
    }
    latent.vars.names <- latent.vars.names[!latent.vars.names %in% re.var]
    fmla <- as.formula(
      object = paste0(
        " ~ ", paste(latent.vars.names, collapse = "+"), glue("+ (1|{re.var})")
      )
    )
    # print(fmla)
    # We need glmer to make this work
    method <-  "glmer" # trying to troubleshoot this - it can clash with the already existing method var in the function call    
    zlmCond <- MAST::zlm(formula = fmla, sca = sca, method = "glmer", ...)
  } else {
    # Original code
    fmla <- as.formula(
      object = paste0(" ~ ", paste(latent.vars.names, collapse = "+"))
    )
    zlmCond <- MAST::zlm(formula = fmla, sca = sca, ...)
  }
  summaryCond <- MAST::summary(object = zlmCond, doLRT = 'conditionGroup2')
  summaryDt <- summaryCond$datatable
  
  # The output format is slightly different, so we need adapt the code
  if(!is.null(re.var)) {
    p_val <- summaryDt[summaryDt$"component" == "H", 4]$`Pr(>Chisq)`
    genes.return <- summaryDt[summaryDt$"component" == "H", 1]$primerid
  } else {
    p_val <- summaryDt[summaryDt[, "component"] == "H", 4]
    genes.return <- summaryDt[summaryDt[, "component"] == "H", 1]
  }
  
  to.return <- data.frame(p_val, row.names = genes.return)
  return(to.return)
}

# This is the zlm function with a minor change to fix a buggy variable being passed possibly due to namespace clash
my_zlm <- function (formula, sca, method = "bayesglm", silent = TRUE, ebayes = TRUE, 
    ebayesControl = NULL, force = FALSE, hook = NULL, parallel = TRUE, 
    LMlike, onlyCoef = FALSE, exprs_values = MAST::assay_idx(sca)$aidx, 
    ...) 
{
    dotsdata = list(...)$data
    if (!is.null(dotsdata)) {
        if (!missing(sca)) 
            stop("Cannot provide both `sca` and `data`")
        sca = dotsdata
    }
    if (!inherits(sca, "SingleCellAssay")) {
        if (inherits(sca, "data.frame")) {
            if (!is.null(dotsdata)) {
                return(.zlm(formula, method = method, silent = silent, 
                  ...))
            }
            else {
                return(.zlm(formula, data = sca, method = method, 
                  silent = silent, ...))
            }
        }
        else {
            stop("`sca` must inherit from `data.frame` or `SingleCellAssay`")
        }
    }
    if (missing(LMlike)) {
        method <- match.arg(method, MAST:::methodDict[, keyword])
        method <- MAST:::methodDict[keyword == method, lmMethod]
        if (!is(sca, "SingleCellAssay")) 
            stop("'sca' must be (or inherit) 'SingleCellAssay'")
        if (!is(formula, "formula")) 
            stop("'formula' must be class 'formula'")
        Formula <- MAST:::removeResponse(formula)
        priorVar <- 1
        priorDOF <- 0
        if (ebayes) {
            # if (!methodDict[lmMethod == method, implementsEbayes]) 
          if (!all(MAST:::methodDict[lmMethod == method, implementsEbayes]))
          stop("Method", method, " does not implement empirical bayes variance shrinkage.")
            ebparm <- MAST:::ebayes(t(SummarizedExperiment:::assay(sca, exprs_values)), ebayesControl, 
                model.matrix(Formula, SummarizedExperiment::colData(sca)))
            priorVar <- ebparm[["v"]]
            priorDOF <- ebparm[["df"]]
            stopifnot(all(!is.na(ebparm)))
        }
        obj <- new_with_repaired_slots(classname = method, design = SummarizedExperiment::colData(sca), 
            formula = Formula, priorVar = priorVar, priorDOF = priorDOF, 
            extra = list(...))
    }
    else {
        if (!missing(formula)) 
            warning("Ignoring formula and using model defined in 'objLMLike'")
        if (!inherits(LMlike, "LMlike")) 
            stop("'LMlike' must inherit from class 'LMlike'")
        obj <- LMlike
    }
    ee <- t(SummarizedExperiment:::assay(sca, exprs_values))
    genes <- colnames(ee)
    ng <- length(genes)
    MM <- MAST:::model.matrix(obj)
    coefNames <- colnames(MM)
    listEE <- setNames(seq_len(ng), genes)
    obj <- MAST:::fit(obj, ee[, 1], silent = silent)
    nerror <- totalerr <- 0
    pb = progress::progress_bar$new(total = ng, format = " Completed [:bar] :percent with :err failures")
    .fitGeneSet <- function(idx) {
        hookOut <- NULL
        tt <- try({
            obj <- MAST:::fit(obj, response = ee[, idx], silent = silent, 
                quick = TRUE)
            if (!is.null(hook)) 
                hookOut <- hook(obj)
            nerror <- 0
        })
        if (is(tt, "try-error")) {
            obj@fitC <- obj@fitD <- NULL
            obj@fitted <- c(C = FALSE, D = FALSE)
            nerror <- nerror + 1
            totalerr = totalerr + 1
            if (nerror > 5 & !force) {
                stop("We seem to be having a lot of problems here...are your tests specified correctly?  \n If you're sure, set force=TRUE.", 
                  tt)
            }
        }
        pb$tick(tokens = list(err = totalerr))
        if (onlyCoef) 
            return(cbind(C = coef(obj, "C"), D = coef(obj, "D")))
        summaries <- MAST:::summarize(obj)
        structure(summaries, hookOut = hookOut)
    }
    if (!parallel || getOption("mc.cores", 1L) == 1) {
        listOfSummaries <- lapply(listEE, .fitGeneSet)
    }
    else {
        listOfSummaries <- parallel::mclapply(listEE, .fitGeneSet, 
            mc.preschedule = TRUE, mc.silent = silent)
    }
    if (onlyCoef) {
        out <- do.call(abind, c(listOfSummaries, rev.along = 0))
        return(aperm(out, c(3, 1, 2)))
    }
    cls <- sapply(listOfSummaries, function(x) class(x))
    complain <- if (force) 
        warning
    else stop
    if (mean(cls == "try-error") > 0.5) 
        complain("Lots of errors here..something is amiss.")
    hookOut <- NULL
    if (!is.null(hook)) 
        hookOut <- lapply(listOfSummaries, attr, which = "hookOut")
    message("\nDone!")
    summaries <- MAST:::collectSummaries(listOfSummaries)
    summaries[["LMlike"]] <- obj
    summaries[["sca"]] <- sca
    summaries[["priorVar"]] <- obj@priorVar
    summaries[["priorDOF"]] <- obj@priorDOF
    summaries[["hookOut"]] <- hookOut
    summaries[["exprs_values"]] <- exprs_values
    summaries[["Class"]] <- "ZlmFit"
    zfit <- do.call(new, as.list(summaries))
    zfit
}

# We will replace the original function with ours inside the Seurat namespace
assignInNamespace('MASTDETest', rem_MASTDETest, asNamespace("Seurat"))
# getFromNamespace("MASTDETest", "Seurat")

assignInNamespace('zlm', my_zlm, asNamespace("MAST"))
# getFromNamespace("zlm", "MAST")
```

# Use modified funcitons to do DE analysis
```{r}
# And now we can call our FindMarkers with random effects model for sample_ID
# Don't forget to include ebayes = F
bcell_deglist_l1<- FindMarkers(bcell_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd4t_deglist_l1<- FindMarkers(cd4t_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd8t_deglist_l1<- FindMarkers(cd8t_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
dc_deglist_l1<- FindMarkers(dc_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
mono_deglist_l1<- FindMarkers(mono_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
nk_deglist_l1<- FindMarkers(nk_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
otherT_deglist_l1<- FindMarkers(otherT_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
other_deglist_l1<- FindMarkers(other_merged_l1,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)


# Predicted celltype L2
#asdc_deglist_l2 <- FindMarkers(asdc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
b_int_deglist_l2 <- FindMarkers(b_int_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
b_mem_deglist_l2 <- FindMarkers(b_mem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
b_naiv_deglist_l2 <- FindMarkers(b_naiv_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
pb_deglist_l2 <- FindMarkers(pb_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd14_mono_deglist_l2 <- FindMarkers(cd14_mono_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd16_mono_deglist_l2 <- FindMarkers(cd16_mono_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd4_ctl_deglist_l2 <- FindMarkers(cd4_ctl_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
# warns on CD4 naive
cd4_naiv_deglist_l2 <- FindMarkers(cd4_naiv_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
#cd4_prolif_deglist_l2 <- FindMarkers(cd4_prolif_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
# warns on CD4 naive
cd4_tcm_deglist_l2 <- FindMarkers(cd4_tcm_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd4_tem_deglist_l2 <- FindMarkers(cd4_tem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd8_naiv_deglist_l2 <- FindMarkers(cd8_naiv_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd8_tcm_deglist_l2 <- FindMarkers(cd8_tcm_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cd8_tem_deglist_l2 <- FindMarkers(cd8_tem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
treg_deglist_l2 <- FindMarkers(cd8_tem_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
gdt_deglist_l2 <- FindMarkers(gdt_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
mait_deglist_l2 <- FindMarkers(mait_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
ilc_deglist_l2 <- FindMarkers(ilc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
#error
#pdc_deglist_l2 <- FindMarkers(pdc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
cdc2_deglist_l2 <- FindMarkers(cdc2_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
nk_deglist_l2 <- FindMarkers(nk_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
nk_cd56br_deglist_l2 <- FindMarkers(nk_cd56br_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
#hspc_deglist_l2 <- FindMarkers(hspc_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
dnt_deglist_l2 <- FindMarkers(dnt_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
nkprolif_deglist_l2 <- FindMarkers(nkprolif_merged_l2,ident.1 = "para", ident.2 = "apara",only.pos = F, test.use = "MAST", assay = "RNA", latent.vars = "Sample.ID",  logfc.threshold = 0.05, re.var = "Sample.ID", ebayes = F, min.pct = 0.1)
```

```{r}
# Concatenate all dataframes of DEGlists together
allcelltypes_exvivo_deglists <- mget(ls(pattern = "*deglist"))

# Save
saveRDS(allcelltypes_exvivo_deglists, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/R21SterileImmunity_All15_samples_CITE_Multiome_allcelltypes_DEGList_Parasitemic_vs_Aparasitemic_test_MAST_with_latent.vars_and RANDOM_EFFECT_of_SampleID_04_25_2023.rds")
```

#GSEA on all celltypes
```{r}
# Optional: Import DEGlists from Google Drive
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("10G1B2gjclPK983F8IH9h47qlGXBzZ79D"), path = temp, overwrite = TRUE)
deglists <- readRDS(file = dl$local_path)

# Deconvolute into individual celltypes' DEGlists
lapply(names(deglists), function(x) assign(x, deglists[[x]], envir = .GlobalEnv))
```

#predicted celltype L1
```{r GSE on predicted celltype L1}
# bcell
bcell_deglist_l1<-bcell_deglist_l1%>%rownames_to_column(., "GeneSymbol")

bcell_rank_l1<-bcell_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_bcell<-NamedGeneRankList2GseaTable(rankedgenes = bcell_rank_l1,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_l1_bcell<- gsea_l1_bcell%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
# cd4t cell
cd4t_deglist_l1<-cd4t_deglist_l1%>%rownames_to_column(., "GeneSymbol")

cd4t_rank_l1<-cd4t_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>% ########### remove infinite values in rank that seem to throw an error in next step
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_cd4t<-NamedGeneRankList2GseaTable(rankedgenes = cd4t_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l1_cd4t<- gsea_l1_cd4t%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# cd8t cell
cd8t_deglist_l1<-cd8t_deglist_l1%>%rownames_to_column(., "GeneSymbol")

cd8t_rank_l1<-cd8t_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_cd8t<-NamedGeneRankList2GseaTable(rankedgenes = cd8t_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l1_cd8t<- gsea_l1_cd8t%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# monocyte
mono_deglist_l1<-mono_deglist_l1%>%rownames_to_column(., "GeneSymbol")

mono_rank_l1<-mono_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_mono<-NamedGeneRankList2GseaTable(rankedgenes = mono_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l1_mono<- gsea_l1_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# dendritic cell
dc_deglist_l1<-dc_deglist_l1%>%rownames_to_column(., "GeneSymbol")

dc_rank_l1<-dc_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_dc<-NamedGeneRankList2GseaTable(rankedgenes = dc_rank_l1,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_l1_dc<- gsea_l1_dc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# nk cell
nk_deglist_l1<-nk_deglist_l1%>%rownames_to_column(., "GeneSymbol")

nk_rank_l1<-nk_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_nk<-NamedGeneRankList2GseaTable(rankedgenes = nk_rank_l1,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_l1_nk<- gsea_l1_nk%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# Other T cell
otherT_deglist_l1<-otherT_deglist_l1%>%rownames_to_column(., "GeneSymbol")

otherT_rank_l1<-otherT_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_otherT<-NamedGeneRankList2GseaTable(rankedgenes = otherT_rank_l1,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_l1_otherT<- gsea_l1_otherT%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# Other
other_deglist_l1<-other_deglist_l1%>%rownames_to_column(., "GeneSymbol")

other_rank_l1<-other_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l1_other<-NamedGeneRankList2GseaTable(rankedgenes = other_rank_l1,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_l1_other<- gsea_l1_other%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()
```

# Concatenate and save GSEA results of each celltype in predictedcelltype L1
```{r concatenate and save GSEA tables for each celltype L1}
# concatenate gsea results
predictedcelltypel1_gsea<- mget(ls(pattern = "gsea_l1*"))

# Save
saveRDS(predictedcelltypel1_gsea, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/R21SterileImmunity_All15_samples_CITE_Multiome_allcelltypesL1_GSEA_Parasitemic_vs_Aparasitemic_MAST_with_latent.vars_and RANDOM_EFFECT_of_SampleID_04_25_2023_TranLab_NamedGeneRankList2GseaTable_02_21_2023.rds")
```

# Run GSEA on each celltype in predictedceltlype L2
```{r GSEA of Predicted celltypes L2}
#ASDC - no ASDcs in aparasitemic kids
#asdc_deglist_l2<-asdc_deglist_l2%>%rownames_to_column(., "GeneSymbol")

#asdc_rank_l2<-asdc_deglist_l2%>%
#  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
#  dplyr::select(GeneSymbol,rankmetric) %>%
#  na.omit() %>%
#  distinct() %>%
#  group_by(GeneSymbol) %>%
#  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
#  arrange(desc(rankmetric)) %>%
#  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

#gsea_l2_asdc<-NamedGeneRankList2GseaTable(rankedgenes = asdc_rank_l2,
#                                       geneset = "all",
#                                       output_directory = tempdir(),
#                                      filename_prefix = "GSEA",
#                                       sampleSize = 101,
#                                       minSize = 20,
#                                       maxSize = Inf,
#                                       scoreType = "std")

#gsea_l2_asdc<- gsea_l2_asdc%>%
#  as_tibble() %>%
#  arrange(desc(NES)) %>% 
#  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
#  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
#  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
#  arrange(padj)%>%
  #filter(padj < 0.20) %>%
#  filter(!grepl("TBA", pathway)) %>%
#  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
#  mutate(neglogpadj = -log10(padj)) %>%
#  mutate(pathway = gsub("gd", "", pathway)) %>%
#  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
#  mutate(pathway = gsub("Vg", "V", pathway)) %>%
#  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
#  mutate(pathway = gsub("_", " ", pathway)) %>%
#  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
#  group_by(module_type) %>%
#  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
#  ungroup() %>%
#  filter(!grepl("TBD", pathway)) %>%
#  arrange(desc(neglogpadj)) %>%
#  droplevels()

##########################################################################################################################################
#b_int
b_int_deglist_l2<-b_int_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_int_rank_l2<-b_int_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_b_int<-NamedGeneRankList2GseaTable(rankedgenes = b_int_rank_l2,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_l2_b_int<- gsea_l2_b_int%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#b_mem
b_mem_deglist_l2<-b_mem_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_mem_rank_l2<-b_mem_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_b_mem<-NamedGeneRankList2GseaTable(rankedgenes = b_mem_rank_l2,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_l2_b_mem<- gsea_l2_b_mem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#b_naiv
b_naiv_deglist_l2<-b_naiv_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_naiv_rank_l2<-b_naiv_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_b_naiv<-NamedGeneRankList2GseaTable(rankedgenes = b_naiv_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_l2_b_naiv<- gsea_l2_b_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#pb
pb_deglist_l2<-pb_deglist_l2%>%rownames_to_column(., "GeneSymbol")

pb_rank_l2<-pb_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_pb<-NamedGeneRankList2GseaTable(rankedgenes = pb_rank_l2,
                                     geneset = "all",
                                     output_directory = tempdir(),
                                     filename_prefix = "GSEA",
                                     sampleSize = 101,
                                     minSize = 20,
                                     maxSize = Inf,
                                     scoreType = "std")

gsea_l2_pb<- gsea_l2_pb%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd14_mono
cd14_mono_deglist_l2<-cd14_mono_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd14_mono_rank_l2<-cd14_mono_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd14_mono<-NamedGeneRankList2GseaTable(rankedgenes = cd14_mono_rank_l2,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_l2_cd14_mono<- gsea_l2_cd14_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd16_mono
cd16_mono_deglist_l2<-cd16_mono_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd16_mono_rank_l2<-cd16_mono_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd16_mono<-NamedGeneRankList2GseaTable(rankedgenes = cd16_mono_rank_l2,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_l2_cd16_mono<- gsea_l2_cd16_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_ctl
cd4_ctl_deglist_l2<-cd4_ctl_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_ctl_rank_l2<-cd4_ctl_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd4_ctl<-NamedGeneRankList2GseaTable(rankedgenes = cd4_ctl_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l2_cd4_ctl<- gsea_l2_cd4_ctl%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_naiv
cd4_naiv_deglist_l2<-cd4_naiv_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_naiv_rank_l2<-cd4_naiv_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd4_naiv<-NamedGeneRankList2GseaTable(rankedgenes = cd4_naiv_rank_l2,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_l2_cd4_naiv<- gsea_l2_cd4_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_prolif
cd4_prolif_deglist_l2<-cd4_prolif_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_prolif_rank_l2<-cd4_prolif_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd4_prolif<-NamedGeneRankList2GseaTable(rankedgenes = cd4_prolif_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_l2_cd4_prolif<- gsea_l2_cd4_prolif%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_tcm
cd4_tcm_deglist_l2<-cd4_tcm_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_tcm_rank_l2<-cd4_tcm_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd4_tcm<-NamedGeneRankList2GseaTable(rankedgenes = cd4_tcm_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l2_cd4_tcm<- gsea_l2_cd4_tcm%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_tem
cd4_tem_deglist_l2<-cd4_tem_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_tem_rank_l2<-cd4_tem_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd4_tem<-NamedGeneRankList2GseaTable(rankedgenes = cd4_tem_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l2_cd4_tem<- gsea_l2_cd4_tem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_naiv
cd8_naiv_deglist_l2<-cd8_naiv_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_naiv_rank_l2<-cd8_naiv_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd8_naiv<-NamedGeneRankList2GseaTable(rankedgenes = cd8_naiv_rank_l2,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_l2_cd8_naiv<- gsea_l2_cd8_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_tcm
cd8_tcm_deglist_l2<-cd8_tcm_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_tcm_rank_l2<-cd8_tcm_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd8_tcm<-NamedGeneRankList2GseaTable(rankedgenes = cd8_tcm_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l2_cd8_tcm<- gsea_l2_cd8_tcm%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_tem
cd8_tem_deglist_l2<-cd8_tem_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_tem_rank_l2<-cd8_tem_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cd8_tem<-NamedGeneRankList2GseaTable(rankedgenes = cd8_tem_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_l2_cd8_tem<- gsea_l2_cd8_tem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#treg
treg_deglist_l2<-treg_deglist_l2%>%rownames_to_column(., "GeneSymbol")

treg_rank_l2<-treg_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_treg<-NamedGeneRankList2GseaTable(rankedgenes = treg_rank_l2,
                                       geneset = "all",
                                       output_directory = tempdir(),
                                       filename_prefix = "GSEA",
                                       sampleSize = 101,
                                       minSize = 20,
                                       maxSize = Inf,
                                       scoreType = "std")

gsea_l2_treg<- gsea_l2_treg%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#gdt
gdt_deglist_l2<-gdt_deglist_l2%>%rownames_to_column(., "GeneSymbol")

gdt_rank_l2<-gdt_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_gdt<-NamedGeneRankList2GseaTable(rankedgenes = gdt_rank_l2,
                                      geneset = "all",
                                      output_directory = tempdir(),
                                      filename_prefix = "GSEA",
                                      sampleSize = 101,
                                      minSize = 20,
                                      maxSize = Inf,
                                      scoreType = "std")

gsea_l2_gdt<- gsea_l2_gdt%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#mait
mait_deglist_l2<-mait_deglist_l2%>%rownames_to_column(., "GeneSymbol")

mait_rank_l2<-mait_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_mait<-NamedGeneRankList2GseaTable(rankedgenes = mait_rank_l2,
                                       geneset = "all",
                                       output_directory = tempdir(),
                                       filename_prefix = "GSEA",
                                       sampleSize = 101,
                                       minSize = 20,
                                       maxSize = Inf,
                                       scoreType = "std")

gsea_l2_mait<- gsea_l2_mait%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#ilc
ilc_deglist_l2<-ilc_deglist_l2%>%rownames_to_column(., "GeneSymbol")

ilc_rank_l2<-ilc_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_ilc<-NamedGeneRankList2GseaTable(rankedgenes = ilc_rank_l2,
                                      geneset = "all",
                                      output_directory = tempdir(),
                                      filename_prefix = "GSEA",
                                      sampleSize = 101,
                                      minSize = 20,
                                      maxSize = Inf,
                                      scoreType = "std")

gsea_l2_ilc<- gsea_l2_ilc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#pdc
pdc_deglist_l2<-pdc_deglist_l2%>%rownames_to_column(., "GeneSymbol")

pdc_rank_l2<-pdc_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_pdc<-NamedGeneRankList2GseaTable(rankedgenes = pdc_rank_l2,
                                      geneset = "all",
                                      output_directory = tempdir(),
                                      filename_prefix = "GSEA",
                                      sampleSize = 101,
                                      minSize = 20,
                                      maxSize = Inf,
                                      scoreType = "std")

gsea_l2_pdc<- gsea_l2_pdc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cdc2
cdc2_deglist_l2<-cdc2_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cdc2_rank_l2<-cdc2_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_cdc2<-NamedGeneRankList2GseaTable(rankedgenes = cdc2_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_l2_cdc2<- gsea_l2_cdc2%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################

##########################################################################################################################################
#nk
nk_deglist_l2<-nk_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nk_rank_l2<-nk_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_nk<-NamedGeneRankList2GseaTable(rankedgenes = nk_rank_l2,
                                     geneset = "all",
                                     output_directory = tempdir(),
                                     filename_prefix = "GSEA",
                                     sampleSize = 101,
                                     minSize = 20,
                                     maxSize = Inf,
                                     scoreType = "std")

gsea_l2_nk<- gsea_l2_nk%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#nk_cd56br
nk_cd56br_deglist_l2<-nk_cd56br_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nk_cd56br_rank_l2_cd56br<-nk_cd56br_deglist_l2%>%
  mutate(rank_cd56brmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rank_cd56brmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rank_cd56brmetric = mean(rank_cd56brmetric)) %>%
  arrange(desc(rank_cd56brmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_nk_cd56br<-NamedGeneRankList2GseaTable(rankedgenes = nk_cd56br_rank_l2_cd56br,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_l2_nk_cd56br<- gsea_l2_nk_cd56br%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#nkprolif
nkprolif_deglist_l2<-nkprolif_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nkprolif_rank_l2prolif<-nkprolif_deglist_l2%>%
  mutate(rankprolifmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankprolifmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankprolifmetric = mean(rankprolifmetric)) %>%
  arrange(desc(rankprolifmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_nkprolif<-NamedGeneRankList2GseaTable(rankedgenes = nkprolif_rank_l2prolif,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_l2_nkprolif<- gsea_l2_nkprolif%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
 # mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

#Untitled 4
##########################################################################################################################################
#hspc
hspc_deglist_l2<-hspc_deglist_l2%>%rownames_to_column(., "GeneSymbol")

hspc_rahspc<-hspc_deglist_l2%>%
  mutate(rahspcmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rahspcmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rahspcmetric = mean(rahspcmetric)) %>%
  arrange(desc(rahspcmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_l2_hspc<-NamedGeneRankList2GseaTable(rankedgenes = hspc_rahspc,
                                       geneset = "all",
                                       output_directory = tempdir(),
                                       filename_prefix = "GSEA",
                                       sampleSize = 101,
                                       minSize = 20,
                                       maxSize = Inf,
                                       scoreType = "std")

gsea_l2_hspc<- gsea_l2_hspc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
 # mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

```

# Concatenate and save GSEA results of each celltype in predictedcelltype L2
```{r concatenate and save GSEA tables for each celltype L1}
# concatenate gsea results
predictedcelltypel2_gsea<- mget(ls(pattern = "gsea_l2*"))

# Save
saveRDS(predictedcelltypel2_gsea, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/R21SterileImmunity_All15_samples_CITE_Multiome_allcelltypesL2_GSEA_Parasitemic_vs_Aparasitemic_MAST_with_latent.vars_and RANDOM_EFFECT_of_SampleID_04_25_2023_TranLab_NamedGeneRankList2GseaTable_02_21_2023.rds")
```

# For Single cell signature scoring, save the gene expression matrices of each celltype's merged object in the SingleCellSignatureExplorer/SingleCellSignatureScorer/data folder
```{r}
#Predicted celltype l1
DefaultAssay(bcell_merged_l1)<-"RNA"
bcell_merged_l1_mat<-t(as.matrix(GetAssayData(bcell_merged_l1)))
write.table(bcell_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_bcell_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd4t_merged_l1)<-"RNA"
cd4t_merged_l1_mat<-t(as.matrix(GetAssayData(cd4t_merged_l1)))
write.table(cd4t_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd4t_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd8t_merged_l1)<-"RNA"
cd8t_merged_l1_mat<-t(as.matrix(GetAssayData(cd8t_merged_l1)))
write.table(cd8t_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd8t_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(dc_merged_l1)<-"RNA"
dc_merged_l1_mat<-t(as.matrix(GetAssayData(dc_merged_l1)))
write.table(dc_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_dc_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(mono_merged_l1)<-"RNA"
mono_merged_l1_mat<-t(as.matrix(GetAssayData(mono_merged_l1)))
write.table(mono_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_mono_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(nk_merged_l1)<-"RNA"
nk_merged_l1_mat<-t(as.matrix(GetAssayData(nk_merged_l1)))
write.table(nk_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_nk_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(otherT_merged_l1)<-"RNA"
otherT_merged_l1_mat<-t(as.matrix(GetAssayData(otherT_merged_l1)))
write.table(otherT_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_otherT_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(other_merged_l1)<-"RNA"
other_merged_l1_mat<-t(as.matrix(GetAssayData(other_merged_l1)))
write.table(other_merged_l1_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_other_merged_l1_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

# Predicted celltype l2
#DefaultAssay(asdc_merged_l2)<-"RNA"
#asdc_merged_l2_mat<-t(as.matrix(GetAssayData(asdc_merged_l2)))
#write.table(asdc_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_asdc_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(b_int_merged_l2)<-"RNA"
b_int_merged_l2_mat<-t(as.matrix(GetAssayData(b_int_merged_l2)))
write.table(b_int_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_b_int_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(b_mem_merged_l2)<-"RNA"
b_mem_merged_l2_mat<-t(as.matrix(GetAssayData(b_mem_merged_l2)))
write.table(b_mem_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_b_mem_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(b_naiv_merged_l2)<-"RNA"
b_naiv_merged_l2_mat<-t(as.matrix(GetAssayData(b_naiv_merged_l2)))
write.table(b_naiv_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_b_naiv_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)


DefaultAssay(pb_merged_l2)<-"RNA"
pb_merged_l2_mat<-t(as.matrix(GetAssayData(pb_merged_l2)))
write.table(pb_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_pb_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd14_mono_merged_l2)<-"RNA"
cd14_mono_merged_l2_mat<-t(as.matrix(GetAssayData(cd14_mono_merged_l2)))
write.table(cd14_mono_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd14_mono_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd16_mono_merged_l2)<-"RNA"
cd16_mono_merged_l2_mat<-t(as.matrix(GetAssayData(cd16_mono_merged_l2)))
write.table(cd16_mono_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd16_mono_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd4_ctl_merged_l2)<-"RNA"
cd4_ctl_merged_l2_mat<-t(as.matrix(GetAssayData(cd4_ctl_merged_l2)))
write.table(cd4_ctl_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd4_ctl_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd4_naiv_merged_l2)<-"RNA"
cd4_naiv_merged_l2_mat<-t(as.matrix(GetAssayData(cd4_naiv_merged_l2)))
write.table(cd4_naiv_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd4_naiv_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd4_prolif_merged_l2)<-"RNA"
cd4_prolif_merged_l2_mat<-t(as.matrix(GetAssayData(cd4_prolif_merged_l2)))
write.table(cd4_prolif_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd4_prolif_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd4_tcm_merged_l2)<-"RNA"
cd4_tcm_merged_l2_mat<-t(as.matrix(GetAssayData(cd4_tcm_merged_l2)))
write.table(cd4_tcm_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd4_tcm_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd4_tem_merged_l2)<-"RNA"
cd4_tem_merged_l2_mat<-t(as.matrix(GetAssayData(cd4_tem_merged_l2)))
write.table(cd4_tem_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd4_tem_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd8_naiv_merged_l2)<-"RNA"
cd8_naiv_merged_l2_mat<-t(as.matrix(GetAssayData(cd8_naiv_merged_l2)))
write.table(cd8_naiv_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd8_naiv_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd8_tcm_merged_l2)<-"RNA"
cd8_tcm_merged_l2_mat<-t(as.matrix(GetAssayData(cd8_tcm_merged_l2)))
write.table(cd8_tcm_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd8_tcm_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cd8_tem_merged_l2)<-"RNA"
cd8_tem_merged_l2_mat<-t(as.matrix(GetAssayData(cd8_tem_merged_l2)))
write.table(cd8_tem_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cd8_tem_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(treg_merged_l2)<-"RNA"
treg_merged_l2_mat<-t(as.matrix(GetAssayData(treg_merged_l2)))
write.table(treg_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_treg_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(gdt_merged_l2)<-"RNA"
gdt_merged_l2_mat<-t(as.matrix(GetAssayData(gdt_merged_l2)))
write.table(gdt_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_gdt_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(mait_merged_l2)<-"RNA"
mait_merged_l2_mat<-t(as.matrix(GetAssayData(mait_merged_l2)))
write.table(mait_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_mait_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(ilc_merged_l2)<-"RNA"
ilc_merged_l2_mat<-t(as.matrix(GetAssayData(ilc_merged_l2)))
write.table(ilc_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_ilc_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(pdc_merged_l2)<-"RNA"
pdc_merged_l2_mat<-t(as.matrix(GetAssayData(pdc_merged_l2)))
write.table(pdc_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_pdc_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(cdc2_merged_l2)<-"RNA"
cdc2_merged_l2_mat<-t(as.matrix(GetAssayData(cdc2_merged_l2)))
write.table(cdc2_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_cdc2_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(nk_merged_l2)<-"RNA"
nk_merged_l2_mat<-t(as.matrix(GetAssayData(nk_merged_l2)))
write.table(nk_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_nk_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(nk_cd56br_merged_l2)<-"RNA"
nk_cd56br_merged_l2_mat<-t(as.matrix(GetAssayData(nk_cd56br_merged_l2)))
write.table(nk_cd56br_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_nk_cd56br_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(hspc_merged_l2)<-"RNA"
hspc_merged_l2_mat<-t(as.matrix(GetAssayData(hspc_merged_l2)))
write.table(hspc_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_hspc_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(dnt_merged_l2)<-"RNA"
dnt_merged_l2_mat<-t(as.matrix(GetAssayData(dnt_merged_l2)))
write.table(dnt_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_dnt_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(nkprolif_merged_l2)<-"RNA"
nkprolif_merged_l2_mat<-t(as.matrix(GetAssayData(nkprolif_merged_l2)))
write.table(nkprolif_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_nkprolif_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

DefaultAssay(platelet_merged_l2)<-"RNA"
platelet_merged_l2_mat<-t(as.matrix(GetAssayData(platelet_merged_l2)))
write.table(platelet_merged_l2_mat, "~/Downloads/SingleCellSignatureExplorer_20220112/SingleCellSignatureScorer/data/R21SterileImm_all_15_samples_exvivo_platelet_merged_l2_integrated_8samples_mat_04_25_2023.tsv", sep="\t", row.names = TRUE, col.names=NA)

```


