---
title: "Integrating in vitro CITESeq and multiome samples"
author: "Prasida Holla"
date: "2023-04-29"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Objective: 
1. Import CITE-seq ex vivo and multiome ex vivo Seurat objects with celltypes annotated based on reference mapping
  - CITE-seq: Samples are integrated as "apara" and "para" objects; split them out based on hashtag signal. Use AddMetaData to add a "Subject.ID" "Age" "Parasitemia" "Sex" columns to be used as co-variates downstream
  - Multiome: Isolate gene expression assays from each sample. Rename new.ident column to "Subject.ID" and add other co-variate columns
  
2. Integrate across all samples using integration anchors. Sanity check: See if old and new predicted celltype annotations match up (How?)

3. Generate uMAP plots at low and high resolution. Sanity check: Check cell type frequencies across samples from this integrated dataset matches with initial frequency assessments

4. Trajectory analysis of ex vivo and in vitro monocytes from unsampled, refmapped seurat objects

5. DGE by MAST and/ or EdgeR using Sample.ID as a random effect

6. GSEA and check against old GSE results

# Load packages
```{r}
library(Seurat) 
library(SeuratWrappers)
library(patchwork)
library(SeuratDisk)
library(tidyverse)
library(googledrive)
library(ggpubr)
library(scales)
library(RColorBrewer)
library(viridis)
library(glue)
library(MAST)
library(monocle3)
```

# Import all datasets to be integrated
```{r import CITE seq and multiome ex vivo datasets, warning=T, results=T}
# CITE-seq ex vivo
# Apara
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("10drXNbsBQU0l1Engwcpuuu7NVedm0h9a"), path = temp, overwrite = TRUE)
apara_c <- readRDS(file = dl$local_path)
#Para
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("10eTZ7O0E6U-MdVVicqgN7nVQoMNEvi3t"), path = temp, overwrite = TRUE)
para_c <- readRDS(file = dl$local_path)

# Multiome
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1QCtC9vhnzkto7IHDO6OVBZjGKvi-9ExG"), path = temp, overwrite = TRUE)
k613_urbc<- readRDS(file = dl$local_path)

temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1QK2FxVqvQk_8yQuBVs3Jzo9CUTcoXLgC"), path = temp, overwrite = TRUE)
k613_irbc<- readRDS(file = dl$local_path)

temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1QKnWlUIx-vIAE8zfnn-1vhFC3JslEX-v"), path = temp, overwrite = TRUE)
k647_urbc<- readRDS(file = dl$local_path)

temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1QW_Br2hX-gsj2VEBM96tpEWwFfBGRlOj"), path = temp, overwrite = TRUE)
k647_irbc<- readRDS(file = dl$local_path)

temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1QsKTELagYLDNANGihD_uYZUAsBDwosRL"), path = temp, overwrite = TRUE)
k651_urbc<- readRDS(file = dl$local_path)

temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1QuMDZFnlMHOpKk3s9Ukswnrkv6vy-4cL"), path = temp, overwrite = TRUE)
k651_irbc<- readRDS(file = dl$local_path)

temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1QyyAwlyzb2M0HeUN7I4VUu102csagBYy"), path = temp, overwrite = TRUE)
k612_urbc<- readRDS(file = dl$local_path)

temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1R1TZxHFIVDv2r9aDE1UE79z7q8PG1EBL"), path = temp, overwrite = TRUE)
k612_irbc<- readRDS(file = dl$local_path)

```

# Seprate out CITE-seq data into individual samples
```{r}
# Apara
rownames(apara_c[["Protein"]])
DefaultAssay(apara_c)<-"Protein"
FeatureScatter(apara_c, feature1 = "S1-uRBC-ly-kali0666-081721", feature2 = "S1-iRBC-ly-kali0666-081721", pt.size = 0.8)+xlim(0,250)+ylim(0,250)
FeatureScatter(apara_c, feature1 = "S4-uRBC-ly-kali0636-081721", feature2 = "S4-iRBC-ly-kali0636-081721", pt.size = 0.8)+xlim(0,250)+ylim(0,250)
k666_urbc<-subset(apara_c,subset =`S1-uRBC-ly-kali0666-081721`>50 & `S1-iRBC-ly-kali0666-081721`<50)
k666_irbc<-subset(apara_c,subset =`S1-uRBC-ly-kali0666-081721`<50 & `S1-iRBC-ly-kali0666-081721`>50)
k636_urbc<-subset(apara_c,subset =`S4-uRBC-ly-kali0636-081721`>50 & `S4-iRBC-ly-kali0636-081721`<50)
k636_irbc<-subset(apara_c,subset =`S4-uRBC-ly-kali0636-081721`<50 & `S4-iRBC-ly-kali0636-081721`>50)

#Para
rownames(para_c[["Protein"]])
DefaultAssay(para_c)<-"Protein"
FeatureScatter(para_c, feature1 = "S6-uRBC-ly-kali0628-081721", feature2 = "S6-iRBC-ly-kali0628-081721", pt.size = 0.8)+xlim(0,250)+ylim(0,250)
FeatureScatter(para_c, feature1 = "S8-uRBC-ly-kali0604-081721", feature2 = "S8-iRBC-ly-kali0604-081721", pt.size = 0.8)+xlim(0,250)+ylim(0,250)
k628_urbc<-subset(para_c,subset =`S6-uRBC-ly-kali0628-081721`>75 & `S6-iRBC-ly-kali0628-081721`<75)
k628_irbc<-subset(para_c,subset =`S6-uRBC-ly-kali0628-081721`<75 & `S6-iRBC-ly-kali0628-081721`>75)
k604_urbc<-subset(para_c,subset =`S8-uRBC-ly-kali0604-081721`>100 & `S8-iRBC-ly-kali0604-081721`<100)
k604_irbc<-subset(para_c,subset =`S8-uRBC-ly-kali0604-081721`<100 & `S8-iRBC-ly-kali0604-081721`>100)
```


# Add a column with unique sample IDs
```{r}
# CITE-seq
ck666ap_uRBC<-AddMetaData(k666_urbc,"Kali0666_Apara_uRBC_Citeseq" , col.name = "Sample.ID")
ck666ap_iRBC<-AddMetaData(k666_irbc,"Kali0666_Apara_iRBC_Citeseq" , col.name = "Sample.ID")
ck636ap_uRBC<-AddMetaData(k636_urbc,"Kali0636_Apara_uRBC_Citeseq" , col.name = "Sample.ID")
ck636ap_iRBC<-AddMetaData(k636_irbc,"Kali0636_Apara_iRBC_Citeseq" , col.name = "Sample.ID")
ck628pa_uRBC<-AddMetaData(k628_urbc,"Kali0628_Para_uRBC_Citeseq" , col.name = "Sample.ID")
ck628pa_iRBC<-AddMetaData(k628_irbc,"Kali0628_Para_iRBC_Citeseq" , col.name = "Sample.ID")
ck604pa_uRBC<-AddMetaData(k604_urbc,"Kali0604_Para_uRBC_Citeseq" , col.name = "Sample.ID")
ck604pa_iRBC<-AddMetaData(k604_irbc,"Kali0604_Para_iRBC_Citeseq" , col.name = "Sample.ID")
# Multiome
mk613ap_urbc<-AddMetaData(k613_urbc,"k613_urbc_Apara_Multiome" , col.name = "Sample.ID")
mk613ap_irbc<-AddMetaData(k613_irbc,"k613_irbc_Apara_Multiome" , col.name = "Sample.ID")
mk647ap_urbc<-AddMetaData(k647_urbc,"k647_urbc_Apara_Multiome" , col.name = "Sample.ID")
mk647ap_irbc<-AddMetaData(k647_irbc,"k647_irbc_Apara_Multiome" , col.name = "Sample.ID")
mk651pa_urbc<-AddMetaData(k651_urbc,"k651_urbc_Para_Multiome" , col.name = "Sample.ID") 
mk651pa_irbc<-AddMetaData(k651_irbc,"k651_irbc_Para_Multiome" , col.name = "Sample.ID")
mk612pa_urbc<-AddMetaData(k612_urbc,"k612_uRBC_Para_Multiome" , col.name = "Sample.ID")
mk612pa_irbc<-AddMetaData(k612_irbc,"k612_iRBC_Para_Multiome" , col.name = "Sample.ID")
```

# Add a sttus and stim column with aparasitemic and parasitemic annotations
```{r}
#Add status column
#CITEseq
ck666ap_uRBC$status = factor("Aparasitemic")
ck666ap_iRBC$status = factor("Aparasitemic")
ck636ap_uRBC$status = factor("Aparasitemic")
ck636ap_iRBC$status = factor("Aparasitemic")
ck628pa_uRBC$status = factor("Parasitemic")
ck628pa_iRBC$status = factor("Parasitemic")
ck604pa_uRBC$status = factor("Parasitemic")
ck604pa_iRBC$status = factor("Parasitemic")
# Multiome
mk613ap_urbc$status = factor("Aparasitemic")
mk613ap_irbc$status = factor("Aparasitemic")
mk647ap_urbc$status = factor("Aparasitemic")
mk647ap_irbc$status = factor("Aparasitemic")
mk651pa_urbc$status = factor("Parasitemic")
mk651pa_irbc$status = factor("Parasitemic")
mk612pa_urbc$status = factor("Parasitemic")
mk612pa_irbc$status = factor("Parasitemic")

#Add stim column 
#CITEseq
ck666ap_uRBC$stim = factor("uRBC")
ck666ap_iRBC$stim = factor("iRBC")
ck636ap_uRBC$stim = factor("uRBC")
ck636ap_iRBC$stim = factor("iRBC")
ck628pa_uRBC$stim = factor("uRBC")
ck628pa_iRBC$stim = factor("iRBC")
ck604pa_uRBC$stim = factor("uRBC")
ck604pa_iRBC$stim = factor("iRBC")
# Multiome
mk613ap_urbc$stim = factor("uRBC")
mk613ap_irbc$stim = factor("iRBC")
mk647ap_urbc$stim = factor("uRBC")
mk647ap_irbc$stim = factor("iRBC")
mk651pa_urbc$stim = factor("uRBC")
mk651pa_irbc$stim = factor("iRBC")
mk612pa_urbc$stim = factor("uRBC")
mk612pa_irbc$stim = factor("iRBC")
```

# Multiome samples have higher mitochondrial gene content and need to be pre-filtered prior to intgration with CITE-seq data
```{r}


mk613ap_urbc<- subset(
  x = mk613ap_urbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)

mk613ap_irbc<- subset(
  x = mk613ap_irbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)

mk647ap_urbc<- subset(
  x = mk647ap_urbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)

mk647ap_irbc<- subset(
  x = mk647ap_irbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)

mk651pa_urbc<- subset(
  x = mk651pa_urbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)

mk651pa_irbc<- subset(
  x = mk651pa_irbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)

mk612pa_urbc<- subset(
  x = mk612pa_urbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)

mk612pa_irbc<- subset(
  x = mk612pa_irbc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 20
)


```

# Equalize cell numbers across all samples by random sampling of larger ones
```{r}
#CITEseq
ck666ap_uRBC #3191
ck666ap_iRBC #3344
ck636ap_uRBC #2272
ck636ap_iRBC #2482
ck628pa_uRBC #3098
ck628pa_iRBC #3733
ck604pa_uRBC #964
ck604pa_iRBC #1908
# Multiome
mk613ap_urbc #4283 cells
mk613ap_irbc #2165 cells
mk647ap_urbc #1830 cells
mk647ap_irbc #1001 cells
mk651pa_urbc #1831 cells
mk651pa_irbc #1014 cells
mk612pa_urbc #3263 cells
mk612pa_irbc #2165 cells


# sample ck604pa_uRBC has least number of cells, which has 964 cells. We will reduce all others to 1431 cells (will make computation of the integrated object easier but is a massive loss of cells)
set.seed(1000) # check with different seeds
ck666ap_uRBC<- ck666ap_uRBC[, sample(colnames(ck666ap_uRBC), size = ncol(ck604pa_uRBC), replace=F)]
ck666ap_iRBC<- ck666ap_iRBC[, sample(colnames(ck666ap_iRBC), size = ncol(ck604pa_uRBC), replace=F)]
ck636ap_uRBC<- ck636ap_uRBC[, sample(colnames(ck636ap_uRBC), size = ncol(ck604pa_uRBC), replace=F)]
ck636ap_iRBC<- ck636ap_iRBC[, sample(colnames(ck636ap_iRBC), size = ncol(ck604pa_uRBC), replace=F)]
ck628pa_uRBC<- ck628pa_uRBC[, sample(colnames(ck628pa_uRBC), size = ncol(ck604pa_uRBC), replace=F)]
ck628pa_iRBC<- ck628pa_iRBC[, sample(colnames(ck628pa_iRBC), size = ncol(ck604pa_uRBC), replace=F)]
ck604pa_iRBC<- ck604pa_iRBC[, sample(colnames(ck604pa_iRBC), size = ncol(ck604pa_uRBC), replace=F)]
mk613ap_urbc<- mk613ap_urbc[, sample(colnames(mk613ap_urbc), size = ncol(ck604pa_uRBC), replace=F)]
mk613ap_irbc<- mk613ap_irbc[, sample(colnames(mk613ap_irbc), size = ncol(ck604pa_uRBC), replace=F)]
mk647ap_urbc<- mk647ap_urbc[, sample(colnames(mk647ap_urbc), size = ncol(ck604pa_uRBC), replace=F)]
mk647ap_irbc<- mk647ap_irbc[, sample(colnames(mk647ap_irbc), size = ncol(ck604pa_uRBC), replace=F)]
mk651pa_urbc<- mk651pa_urbc[, sample(colnames(mk651pa_urbc), size = ncol(ck604pa_uRBC), replace=F)]
mk651pa_irbc<- mk651pa_irbc[, sample(colnames(mk651pa_irbc), size = ncol(ck604pa_uRBC), replace=F)]
mk612pa_urbc<- mk612pa_urbc[, sample(colnames(mk612pa_urbc), size = ncol(ck604pa_uRBC), replace=F)]
mk612pa_irbc<- mk612pa_irbc[, sample(colnames(mk612pa_irbc), size = ncol(ck604pa_uRBC), replace=F)]
```

# For this purpose, we only want RNA assays
```{r}
DefaultAssay(ck666ap_uRBC)<-"RNA"
DefaultAssay(ck666ap_iRBC)<-"RNA"
DefaultAssay(ck636ap_uRBC)<-"RNA"
DefaultAssay(ck636ap_iRBC)<-"RNA"
DefaultAssay(ck628pa_uRBC)<-"RNA"
DefaultAssay(ck628pa_iRBC)<-"RNA"
DefaultAssay(ck604pa_uRBC)<-"RNA"
DefaultAssay(ck604pa_iRBC)<-"RNA"
DefaultAssay(mk613ap_urbc)<-"RNA"
DefaultAssay(mk613ap_irbc)<-"RNA"
DefaultAssay(mk647ap_urbc)<-"RNA"
DefaultAssay(mk647ap_irbc)<-"RNA"
DefaultAssay(mk651pa_urbc)<-"RNA"
DefaultAssay(mk651pa_irbc)<-"RNA"
DefaultAssay(mk612pa_urbc)<-"RNA"
DefaultAssay(mk612pa_irbc)<-"RNA"
```

# Integrate all 15 samples (except Kali0613, which has not been seqeunced yet)
```{r}
# Step 1: Find integration anchors across each of the objects and integrate
all<-merge(ck666ap_uRBC, c(ck666ap_iRBC,
                           ck636ap_uRBC,
                           ck636ap_iRBC,
                           ck628pa_uRBC,
                           ck628pa_iRBC,
                           ck604pa_uRBC,
                           ck604pa_iRBC,
                           mk613ap_urbc,
                           mk613ap_irbc,
                           mk647ap_urbc,
                           mk647ap_irbc,
                           mk651pa_urbc,
                           mk651pa_irbc,
                           mk612pa_urbc,
                           mk612pa_irbc),
           add.cell.ids = c("Kali0666_uRBC_Apara_Citeseq",
                            "Kali0666_iRBC_Apara_Citeseq",
                            "Kali0636_uRBC_Apara_Citeseq",
                            "Kali0636_iRBC_Apara_Citeseq",
                            "Kali0628_uRBC_Para_Citeseq",
                            "Kali0628_iRBC_Para_Citeseq",
                            "Kali0604_uRBC_Para_Citeseq",
                            "Kali0604_iRBC_Para_Citeseq",
                           "Kali0613_uRBC_Apara_Multiome",
                            "Kali0613_iRBC_Apara_Multiome",
                            "Kali0647_uRBC_Apara_Multiome",
                            "Kali0647_iRBC_Apara_Multiome",
                            "Kali0651_uRBC_Para_Multiome",
                            "Kali0651_iRBC_Para_Multiome",
                            "Kali0612_uRBC_Para_Multiome",
                            "Kali0612_iRBC_Para_Multiome"))

all.list <- SplitObject(all, split.by = "Sample.ID")  

for (i in 1:length(all.list)) {
  all.list[[i]] <- NormalizeData(all.list[[i]], verbose = FALSE)
  all.list[[i]] <- FindVariableFeatures(all.list[[i]], selection.method = "vst",
                                           nfeatures = 2000,verbose = FALSE)
}

all.anchors<-FindIntegrationAnchors(all.list, dim=1:50,anchor.features = 2000,
                                       l2.norm = T, k.anchor = 5,k.score = 20,
                                       max.features = 200)

all.integrated <- IntegrateData(anchorset = all.anchors, dims = 1:50)

# Step 2: Reduce and run dimensionality reduction

DefaultAssay(all.integrated) <- "integrated"
all.integrated <- ScaleData(all.integrated, verbose = FALSE)
all.integrated <- RunPCA(all.integrated, npcs = 50, verbose = FALSE)

# Check optimal number of PCs/ dimensions to be included for dim.reduction. There are several ways to do this
# 1. Identify the top 10-50 most highly variable genes. This approach is more useful for a relatively pure celltype than a mixture of celltypes such as PBMC
top50 <- head(VariableFeatures(all.integrated), 50)

# 2. Elbow plot of PCs vs variance
ElbowPlot(all.integrated) # shows that beyond 6 PCs, variance drops below 2 and this can be a cutoff

# 3. Heatmap of PCs to see where variance is breaking down (looks to be ~ 6PCs)
#DimHeatmap(all.integrated, dims = 1:20, cells = 500, balanced = TRUE)

# Based on these, we will use 6 PCs for uMAP and tSNE analysis
all.integrated <- RunUMAP(all.integrated, reduction = "pca", dims = 1:15)
all.integrated <- RunTSNE(all.integrated, reduction = "pca", dims = 1:15, check_duplicates=F)

all.integrated <- FindNeighbors(all.integrated, reduction = "pca", 
                                   dims = 1:15)
all.integrated <- FindClusters(all.integrated, resolution = 0.5)

#Step 3a: Load the reference dataset (note: this is hard to upload from google Drive dur to its large size)
reference <- LoadH5Seurat("~/Desktop/R21 Sterile Immunity data and scripts/CITE-seq/pbmc_multimodal.h5seurat")

# Step 3b: SCTransform the query datasets
#Set default assay to RNA again (not sure why that should be needed, throws error if you do not
#Error in make_cell_attr(umi, cell_attr, latent_var, batch_var, latent_var_nonreg,  :  count matrix must have row and column names)
DefaultAssay(all.integrated)<-"RNA"
all_sc<- SCTransform(all.integrated, verbose = FALSE) 

# Step 3c: Find anchors between reference and query datasets
all.anchors<- FindTransferAnchors(
  reference = reference,
  query = all_sc, 
  normalization.method = "SCT",
  reference.reduction = "spca",
  dims = 1:50
)

# Step 3d: Run MapQuery, which transfers cell type labels and protein data from reference to query
#5y
refmapped.all <- MapQuery(
  anchorset = all.anchors,
  query = all_sc,
  reference = reference,
  refdata = list(
    celltype.l1 = "celltype.l1",
    celltype.l2 = "celltype.l2",
    predicted_ADT = "ADT"
  ),
  reference.reduction = "spca", 
  reduction.model = "wnn.umap"
)

saveRDS(refmapped.all, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/In vitro/Refmapped_and_unsampled_invitro_CITEseq_and_multiome_uRBC_andiRBC_05012023.rds")

# Import here from Google Drive
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1_BU-rYHNbPOAy2V-MyZS5Yz9lCn-U9_q"), path = temp, overwrite = TRUE)
refmapped.all<- readRDS(file = dl$local_path)
```

# Trajectory analaysis of ex vivo and in vitro monocytes
```{r}
# Import ex vivo unsampled seurat object here from Google Drive
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id("1ctsDxgIUaNmgJ1rGPKHh2Y10RqzosUnJ"), path = temp, overwrite = TRUE)
refmapped.all_exvivo<- readRDS(file = dl$local_path)

mono<-subset(refmapped.all_exvivo, subset=predicted.celltype.l1=="Mono")

mono.split<-SplitObject(mono, split.by = "batch")
mono_para<-mono.split$Parasitemic
mono_apara<-mono.split$Aparasitemic

cdsp<-as.cell_data_set(mono_para)
cdsap<-as.cell_data_set(mono_apara)


cdsp <- cluster_cells(cds = cdsp, reduction_method = "UMAP")
cdsp <- learn_graph(cdsp, use_partition = TRUE)

cdsp <- order_cells(cdsp, reduction_method = "UMAP", root_cells = NULL)

mono_para <- AddMetaData(
  object = mono_para,
  metadata = cdsp@principal_graph_aux@listData$UMAP$pseudotime,
  col.name = "pseudo_traj"
)

para_plot_traj<-FeaturePlot(mono_para, "pseudo_traj", pt.size = 0.1, reduction = "umap") + scale_color_viridis_c() +NoLegend() +
  ggtitle("Parasitemic_exvivo_pseudotime_trajectory")+
  theme(axis.text.y = element_text(size=14),
        axis.text.x = element_text(size=14),
        axis.title.y  = element_text(size=14),
        axis.title.x  = element_text(size=14),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "darkgrey", fill=NA, size=1),
        plot.margin=unit(c(0.01,0.01,0.01,0.01), "null"),
        legend.position = "none")+
  grids(linetype = "dashed")

cdsap <- cluster_cells(cds = cdsap, reduction_method = "UMAP")
cdsap <- learn_graph(cdsap, use_partition = TRUE)
# here I used the uninfected day 4 cells as root, which gave some confusing results
cdsap <- order_cells(cdsap, reduction_method = "UMAP", root_cells = NULL)

mono_apara <- AddMetaData(
  object = mono_apara,
  metadata = cdsap@principal_graph_aux@listData$UMAP$pseudotime,
  col.name = "pseudo_traj"
)

apara_plot_traj<-FeaturePlot(mono_apara, "pseudo_traj", pt.size = 0.1, reduction = "umap")+ scale_color_viridis_c()+NoLegend() +
  ggtitle("Aprasitemic_pseudotime_trajectory")+
  theme(axis.text.y = element_text(size=14),
        axis.text.x = element_text(size=14),
        axis.title.y  = element_text(size=14),
        axis.title.x  = element_text(size=14),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "darkgrey", fill=NA, size=1),
        plot.margin=unit(c(0.01,0.01,0.01,0.01), "null"),
        legend.position = "none")+
  grids(linetype = "dashed")


pdf("/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/Exvivo_all15samples_minusk613_unsampled_pseudotime_trajectories_of_monocytes.pdf", height = 5, width = 10)
apara_plot_traj|para_plot_traj
dev.off()

# pseudotime of in vitro stimmed cells

# Import unsampled in vitro object here from Google Drive
#this link needs to be updated
refmapped.invitro<-readRDS("/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/In vitro/R21SterileImm_invitro_all8_CITEseq_and_multiome_integrated.rds")

#temp <- tempfile(fileext = ".rds")
#dl <- drive_download(
#  as_id(""), path = temp, overwrite = TRUE)
#refmapped.all<- readRDS(file = dl$local_path)

mono<-subset(refmapped.invitro, subset=predicted.celltype.l1=="Mono")

mono.split<-SplitObject(mono, split.by = "status")
mono_para<-mono.split$Parasitemic
mono_apara<-mono.split$Aparasitemic

mono_para<-SplitObject(mono_para, split.by = "stim")
mono_para_urbc<-mono_para$uRBC
mono_para_irbc<-mono_para$iRBC

mono_apara<-SplitObject(mono_apara, split.by = "stim")
mono_apara_urbc<-mono_apara$uRBC
mono_apara_irbc<-mono_apara$iRBC


#CDS
cdspu<-as.cell_data_set(mono_para_urbc)
cdspi<-as.cell_data_set(mono_para_irbc)
cdsapu<-as.cell_data_set(mono_apara_urbc)
cdsapi<-as.cell_data_set(mono_apara_irbc)

#Para_urbc
cdspu <- cluster_cells(cds = cdspu, reduction_method = "UMAP")
cdspu <- learn_graph(cdspu, use_partition = TRUE)
cdspu <- order_cells(cdspu, reduction_method = "UMAP", root_cells = NULL)

mono_para_urbc <- AddMetaData(
  object = mono_para_urbc,
  metadata = cdspu@principal_graph_aux@listData$UMAP$pseudotime,
  col.name = "pseudo_traj"
)

para_urbc_plot_traj<-FeaturePlot(mono_para_urbc, "pseudo_traj", pt.size = 1, reduction = "umap", keep.scale = "feature") + scale_color_viridis_c() +
  ggtitle("Parasitemic uRBC")+
  theme(axis.text.y = element_text(size=14),
        axis.text.x = element_text(size=14),
        axis.title.y  = element_text(size=14),
        axis.title.x  = element_text(size=14),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "darkgrey", fill=NA, size=1),
        plot.margin=unit(c(0.01,0.01,0.01,0.01), "null"),
        legend.position = "none")+
  grids(linetype = "dashed")

#Para_irbc
cdspi <- cluster_cells(cds = cdspi, reduction_method = "UMAP")
cdspi <- learn_graph(cdspi, use_partition = TRUE)
cdspi <- order_cells(cdspi, reduction_method = "UMAP", root_cells = NULL)

mono_para_irbc <- AddMetaData(
  object = mono_para_irbc,
  metadata = cdspi@principal_graph_aux@listData$UMAP$pseudotime,
  col.name = "pseudo_traj"
)

para_irbc_plot_traj<-FeaturePlot(mono_para_irbc, "pseudo_traj", pt.size = 1, reduction = "umap", keep.scale = "feature") + scale_color_viridis_c() +
  ggtitle("Parasitemic iRBC")+
  theme(axis.text.y = element_text(size=14),
        axis.text.x = element_text(size=14),
        axis.title.y  = element_text(size=14),
        axis.title.x  = element_text(size=14),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "darkgrey", fill=NA, size=1),
        plot.margin=unit(c(0.01,0.01,0.01,0.01), "null"),
        legend.position = "none")+
  grids(linetype = "dashed")

#apara_urbc
cdsapu <- cluster_cells(cds = cdsapu, reduction_method = "UMAP")
cdsapu <- learn_graph(cdsapu, use_partition = TRUE)
cdsapu <- order_cells(cdsapu, reduction_method = "UMAP", root_cells = NULL)

mono_apara_urbc <- AddMetaData(
  object = mono_apara_urbc,
  metadata = cdsapu@principal_graph_aux@listData$UMAP$pseudotime,
  col.name = "pseudo_traj"
)

apara_urbc_plot_traj<-FeaturePlot(mono_apara_urbc, "pseudo_traj", pt.size = 1, reduction = "umap", keep.scale = "feature") + scale_color_viridis_c() +
  ggtitle("Aparasitemic uRBC")+
  theme(axis.text.y = element_text(size=14),
        axis.text.x = element_text(size=14),
        axis.title.y  = element_text(size=14),
        axis.title.x  = element_text(size=14),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "darkgrey", fill=NA, size=1),
        plot.margin=unit(c(0.01,0.01,0.01,0.01), "null"),
        legend.position = "none")+
  grids(linetype = "dashed")

#apara_irbc
cdsapi <- cluster_cells(cds = cdsapi, reduction_method = "UMAP")
cdsapi <- learn_graph(cdsapi, use_partition = TRUE)
cdsapi <- order_cells(cdsapi, reduction_method = "UMAP", root_cells = NULL)

mono_apara_irbc <- AddMetaData(
  object = mono_apara_irbc,
  metadata = cdsapi@principal_graph_aux@listData$UMAP$pseudotime,
  col.name = "pseudo_traj"
)

apara_irbc_plot_traj<-FeaturePlot(mono_apara_irbc, "pseudo_traj", pt.size = 1, reduction = "umap", keep.scale = "feature") + scale_color_viridis_c() +
  ggtitle("Aparasitemic iRBC")+
  theme(axis.text.y = element_text(size=14),
        axis.text.x = element_text(size=14),
        axis.title.y  = element_text(size=14),
        axis.title.x  = element_text(size=14),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "darkgrey", fill=NA, size=1),
        plot.margin=unit(c(0.01,0.01,0.01,0.01), "null"),
        legend.position = "none")+
  grids(linetype = "dashed")

pdf("/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/invitro_all15samples_minusk613_unsampled_pseudotime_trajectories_of_monocytes.pdf", height = 10, width = 10)
ggarrange(apara_urbc_plot_traj, apara_irbc_plot_traj, para_urbc_plot_traj, para_irbc_plot_traj, ncol=2, nrow=2)
dev.off()

```

#Seperate out data by status and stim
```{r}
#Status
all.split<-SplitObject(refmapped.all, split.by = "status")
all_para<-all.split$Parasitemic
all_apara<-all.split$Aparasitemic

#Stim
#Para
all_para.split<-SplitObject(all_para, split.by = "stim")
all_para_urbc<-all_para.split$uRBC
all_para_irbc<-all_para.split$iRBC

#Apara
all_apara.split<-SplitObject(all_apara, split.by = "stim")
all_apara_urbc<-all_apara.split$uRBC
all_apara_irbc<-all_apara.split$iRBC
```

Seperate out individual ceclltypes
```{r}
#Para_uRBC
# Low resolution celltypes
list_para_urbc_l1 <- c()
for (i in levels(factor(all_para_urbc[["predicted.celltype.l1"]][[1]]))){
  list_para_urbc_l1[[i]] <- subset(all_para_urbc, subset=`predicted.celltype.l1`==i)
}

# High resolution celltypes
list_para_urbc_l2 <- c()
for (i in levels(factor(all_para_urbc[["predicted.celltype.l2"]][[1]]))){
  list_para_urbc_l2[[i]] <- subset(all_para_urbc, subset=`predicted.celltype.l2`==i)
}

#Para_iRBC
# Low resolution celltypes
list_para_irbc_l1 <- c()
for (i in levels(factor(all_para_irbc[["predicted.celltype.l1"]][[1]]))){
  list_para_irbc_l1[[i]] <- subset(all_para_irbc, subset=`predicted.celltype.l1`==i)
}

# High resolution celltypes
list_para_irbc_l2 <- c()
for (i in levels(factor(all_para_irbc[["predicted.celltype.l2"]][[1]]))){
  list_para_irbc_l2[[i]] <- subset(all_para_irbc, subset=`predicted.celltype.l2`==i)
}

#Apara_uRBC
# Low resolution celltypes
list_apara_urbc_l1 <- c()
for (i in levels(factor(all_apara_urbc[["predicted.celltype.l1"]][[1]]))){
  list_apara_urbc_l1[[i]] <- subset(all_apara_urbc, subset=`predicted.celltype.l1`==i)
}

# High resolution celltypes
list_apara_urbc_l2 <- c()
for (i in levels(factor(all_apara_urbc[["predicted.celltype.l2"]][[1]]))){
  list_apara_urbc_l2[[i]] <- subset(all_apara_urbc, subset=`predicted.celltype.l2`==i)
}

#apara_iRBC
# Low resolution celltypes
list_apara_irbc_l1 <- c()
for (i in levels(factor(all_apara_irbc[["predicted.celltype.l1"]][[1]]))){
  list_apara_irbc_l1[[i]] <- subset(all_apara_irbc, subset=`predicted.celltype.l1`==i)
}

# High resolution celltypes
list_apara_irbc_l2 <- c()
for (i in levels(factor(all_apara_irbc[["predicted.celltype.l2"]][[1]]))){
  list_apara_irbc_l2[[i]] <- subset(all_apara_irbc, subset=`predicted.celltype.l2`==i)
}
```

#split out individual celltypes
```{r}
# Predicted celltype L1

#Para_uRBC
# Check indices prior to assignment for separation of cell types
names(list_para_urbc_l1)
b_para_urbc_l1<-list_para_urbc_l1[[1]]
cd4t_para_urbc_l1<-list_para_urbc_l1[[2]]
cd8t_para_urbc_l1<-list_para_urbc_l1[[3]]
dc_para_urbc_l1<-list_para_urbc_l1[[4]]
mono_para_urbc_l1<-list_para_urbc_l1[[5]]
nk_para_urbc_l1<-list_para_urbc_l1[[6]]
otherT_para_urbc_l1<-list_para_urbc_l1[[8]]
other_para_urbc_l1<-list_para_urbc_l1[[7]]


#Para_irbc
# Check indices prior to assignment for seperation of cell types
names(list_para_irbc_l1)
b_para_irbc_l1<-list_para_irbc_l1[[1]]
cd4t_para_irbc_l1<-list_para_irbc_l1[[2]]
cd8t_para_irbc_l1<-list_para_irbc_l1[[3]]
dc_para_irbc_l1<-list_para_irbc_l1[[4]]
mono_para_irbc_l1<-list_para_irbc_l1[[5]]
nk_para_irbc_l1<-list_para_irbc_l1[[6]]
otherT_para_irbc_l1<-list_para_irbc_l1[[8]]
other_para_irbc_l1<-list_para_irbc_l1[[7]]


#apara_urbc
# Check indices prior to assignment for separation of cell types
names(list_apara_urbc_l1)
b_apara_urbc_l1<-list_apara_urbc_l1[[1]]
cd4t_apara_urbc_l1<-list_apara_urbc_l1[[2]]
cd8t_apara_urbc_l1<-list_apara_urbc_l1[[3]]
#dc_apara_urbc_l1<-list_apara_urbc_l1[[4]]
mono_apara_urbc_l1<-list_apara_urbc_l1[[4]]
nk_apara_urbc_l1<-list_apara_urbc_l1[[5]]
otherT_apara_urbc_l1<-list_apara_urbc_l1[[7]]
other_apara_urbc_l1<-list_apara_urbc_l1[[6]]


#apara_irbc
# Check indices prior to assignment for seperation of cell types
names(list_apara_irbc_l1)
b_apara_irbc_l1<-list_apara_irbc_l1[[1]]
cd4t_apara_irbc_l1<-list_apara_irbc_l1[[2]]
cd8t_apara_irbc_l1<-list_apara_irbc_l1[[3]]
#dc_apara_irbc_l1<-list_apara_irbc_l1[[4]]
mono_apara_irbc_l1<-list_apara_irbc_l1[[4]]
nk_apara_irbc_l1<-list_apara_irbc_l1[[5]]
otherT_apara_irbc_l1<-list_apara_irbc_l1[[7]]
other_apara_irbc_l1<-list_apara_irbc_l1[[6]]

# Predicted celltype L2
#Para uRBC
names(list_para_urbc_l2)
#asdc_par_urbc_l2<-list_para_urbc_l2[[1]]
b_int_par_urbc_l2<-list_para_urbc_l2[[1]]
b_mem_par_urbc_l2<-list_para_urbc_l2[[2]]
b_naiv_par_urbc_l2<-list_para_urbc_l2[[3]]
pb_par_urbc_l2<-list_para_urbc_l2[[19]]
cd14_mono_par_urbc_l2<-list_para_urbc_l2[[4]]
#cd16_mono_par_urbc_l2<-list_para_urbc_l2[[]]
#cd4_ctl_par_urbc_l2<-list_para_urbc_l2[[7]]
cd4_naiv_par_urbc_l2<-list_para_urbc_l2[[5]]
#cd4_prolif_par_urbc_l2<-list_para_urbc_l2[[9]]
cd4_tcm_par_urbc_l2<-list_para_urbc_l2[[6]]
cd4_tem_par_urbc_l2<-list_para_urbc_l2[[7]]
cd8_naiv_par_urbc_l2<-list_para_urbc_l2[[8]]
#cd8_prolif_par_urbc_l2<-list_para_urbc_l2[[]]
cd8_tcm_par_urbc_l2<-list_para_urbc_l2[[9]]
cd8_tem_par_urbc_l2<-list_para_urbc_l2[[10]]
treg_par_urbc_l2<-list_para_urbc_l2[[21]]
gdt_par_urbc_l2<-list_para_urbc_l2[[12]]
mait_par_urbc_l2<-list_para_urbc_l2[[15]]
ilc_par_urbc_l2<-list_para_urbc_l2[[14]]
pdc_par_urbc_l2<-list_para_urbc_l2[[18]]
#cdc2_par_urbc_l2<-list_para_urbc_l2[[]]
nk_par_urbc_l2<-list_para_urbc_l2[[16]]
nk_cd56br_par_urbc_l2<-list_para_urbc_l2[[17]]
hspc_par_urbc_l2<-list_para_urbc_l2[[13]]
dnt_par_urbc_l2<-list_para_urbc_l2[[11]]
#nkprolif_par_urbc_l2<-list_para_urbc_l2[[]]
platelet_par_urbc_l2<-list_para_urbc_l2[[20]]

#Para irbc
names(list_para_irbc_l2)
#asdc_par_irbc_l2<-list_para_irbc_l2[[1]]
b_int_par_irbc_l2<-list_para_irbc_l2[[1]]
b_mem_par_irbc_l2<-list_para_irbc_l2[[2]]
b_naiv_par_irbc_l2<-list_para_irbc_l2[[3]]
pb_par_irbc_l2<-list_para_irbc_l2[[21]]
cd14_mono_par_irbc_l2<-list_para_irbc_l2[[4]]
#cd16_mono_par_irbc_l2<-list_para_irbc_l2[[]]
cd4_ctl_par_irbc_l2<-list_para_irbc_l2[[5]]
cd4_naiv_par_irbc_l2<-list_para_irbc_l2[[6]]
#cd4_prolif_par_irbc_l2<-list_para_irbc_l2[[9]]
cd4_tcm_par_irbc_l2<-list_para_irbc_l2[[7]]
cd4_tem_par_irbc_l2<-list_para_irbc_l2[[8]]
cd8_naiv_par_irbc_l2<-list_para_irbc_l2[[9]]
#cd8_prolif_par_irbc_l2<-list_para_irbc_l2[[]]
cd8_tcm_par_irbc_l2<-list_para_irbc_l2[[10]]
cd8_tem_par_irbc_l2<-list_para_irbc_l2[[11]]
treg_par_irbc_l2<-list_para_irbc_l2[[22]]
gdt_par_irbc_l2<-list_para_irbc_l2[[13]]
mait_par_irbc_l2<-list_para_irbc_l2[[16]]
ilc_par_irbc_l2<-list_para_irbc_l2[[15]]
pdc_par_irbc_l2<-list_para_irbc_l2[[20]]
#cdc2_par_irbc_l2<-list_para_irbc_l2[[]]
nk_par_irbc_l2<-list_para_irbc_l2[[17]]
nk_cd56br_par_irbc_l2<-list_para_irbc_l2[[19]]
hspc_par_irbc_l2<-list_para_irbc_l2[[14]]
dnt_par_irbc_l2<-list_para_irbc_l2[[12]]
nkprolif_par_irbc_l2<-list_para_irbc_l2[[18]]
#platelet_par_irbc_l2<-list_para_irbc_l2[[]]

#apara urbc
names(list_apara_urbc_l2)
#asdc_apar_urbc_l2<-list_apara_urbc_l2[[1]]
b_int_apar_urbc_l2<-list_apara_urbc_l2[[1]]
b_mem_apar_urbc_l2<-list_apara_urbc_l2[[2]]
b_naiv_apar_urbc_l2<-list_apara_urbc_l2[[3]]
#pb_apar_urbc_l2<-list_apara_urbc_l2[[]]
cd14_mono_apar_urbc_l2<-list_apara_urbc_l2[[4]]
cd16_mono_apar_urbc_l2<-list_apara_urbc_l2[[5]]
cd4_ctl_apar_urbc_l2<-list_apara_urbc_l2[[6]]
cd4_naiv_apar_urbc_l2<-list_apara_urbc_l2[[7]]
#cd4_prolif_apar_urbc_l2<-list_apara_urbc_l2[[]]
cd4_tcm_apar_urbc_l2<-list_apara_urbc_l2[[8]]
cd4_tem_apar_urbc_l2<-list_apara_urbc_l2[[9]]
cd8_naiv_apar_urbc_l2<-list_apara_urbc_l2[[10]]
#cd8_prolif_apar_urbc_l2<-list_apara_urbc_l2[[]]
cd8_tcm_apar_urbc_l2<-list_apara_urbc_l2[[11]]
cd8_tem_apar_urbc_l2<-list_apara_urbc_l2[[12]]
treg_apar_urbc_l2<-list_apara_urbc_l2[[22]]
gdt_apar_urbc_l2<-list_apara_urbc_l2[[14]]
mait_apar_urbc_l2<-list_apara_urbc_l2[[17]]
ilc_apar_urbc_l2<-list_apara_urbc_l2[[16]]
#pdc_apar_urbc_l2<-list_apara_urbc_l2[[]]
#cdc2_apar_urbc_l2<-list_apara_urbc_l2[[]]
nk_apar_urbc_l2<-list_apara_urbc_l2[[18]]
nk_cd56br_apar_urbc_l2<-list_apara_urbc_l2[[19]]
hspc_apar_urbc_l2<-list_apara_urbc_l2[[15]]
dnt_apar_urbc_l2<-list_apara_urbc_l2[[13]]
nkprolif_apar_urbc_l2<-list_apara_urbc_l2[[19]]
platelet_apar_urbc_l2<-list_apara_urbc_l2[[21]]

#apara irbc
names(list_apara_irbc_l2)
#asdc_apar_irbc_l2<-list_apara_irbc_l2[[1]]
b_int_apar_irbc_l2<-list_apara_irbc_l2[[1]]
b_mem_apar_irbc_l2<-list_apara_irbc_l2[[2]]
b_naiv_apar_irbc_l2<-list_apara_irbc_l2[[3]]
pb_apar_irbc_l2<-list_apara_irbc_l2[[21]]
cd14_mono_apar_irbc_l2<-list_apara_irbc_l2[[4]]
cd16_mono_apar_irbc_l2<-list_apara_irbc_l2[[5]]
cd4_ctl_apar_irbc_l2<-list_apara_irbc_l2[[6]]
cd4_naiv_apar_irbc_l2<-list_apara_irbc_l2[[7]]
#cd4_prolif_apar_irbc_l2<-list_apara_irbc_l2[[]]
cd4_tcm_apar_irbc_l2<-list_apara_irbc_l2[[8]]
cd4_tem_apar_irbc_l2<-list_apara_irbc_l2[[9]]
cd8_naiv_apar_irbc_l2<-list_apara_irbc_l2[[10]]
#cd8_prolif_apar_irbc_l2<-list_apara_irbc_l2[[]]
cd8_tcm_apar_irbc_l2<-list_apara_irbc_l2[[11]]
cd8_tem_apar_irbc_l2<-list_apara_irbc_l2[[12]]
treg_apar_irbc_l2<-list_apara_irbc_l2[[23]]
gdt_apar_irbc_l2<-list_apara_irbc_l2[[14]]
mait_apar_irbc_l2<-list_apara_irbc_l2[[17]]
ilc_apar_irbc_l2<-list_apara_irbc_l2[[16]]
#pdc_apar_irbc_l2<-list_apara_irbc_l2[[]]
#cdc2_apar_irbc_l2<-list_apara_irbc_l2[[]]
nk_apar_irbc_l2<-list_apara_irbc_l2[[18]]
nk_cd56br_apar_irbc_l2<-list_apara_irbc_l2[[19]]
hspc_apar_irbc_l2<-list_apara_irbc_l2[[15]]
dnt_apar_irbc_l2<-list_apara_irbc_l2[[13]]
nkprolif_apar_irbc_l2<-list_apara_irbc_l2[[19]]
platelet_apar_irbc_l2<-list_apara_irbc_l2[[22]]
```

#Merge parasitemic and aparasitemic objects for each celltype
```{r}
# predicted celltype l1
#Para_uRBC
Idents(b_para_urbc_l1)<-"urbc"
Idents(cd4t_para_urbc_l1)<-"urbc"
Idents(cd8t_para_urbc_l1)<-"urbc"
Idents(dc_para_urbc_l1)<-"urbc"
Idents(mono_para_urbc_l1)<-"urbc"
Idents(nk_para_urbc_l1)<-"urbc"
Idents(otherT_para_urbc_l1)<-"urbc"
Idents(other_para_urbc_l1)<-"urbc"


#Para_irbc
Idents(b_para_irbc_l1)<-"irbc"
Idents(cd4t_para_irbc_l1)<-"irbc"
Idents(cd8t_para_irbc_l1)<-"irbc"
Idents(dc_para_irbc_l1)<-"irbc"
Idents(mono_para_irbc_l1)<-"irbc"
Idents(nk_para_irbc_l1)<-"irbc"
Idents(otherT_para_irbc_l1)<-"irbc"
Idents(other_para_irbc_l1)<-"irbc"


#apara_urbc
Idents(b_apara_urbc_l1)<-"urbc"
Idents(cd4t_apara_urbc_l1)<-"urbc"
Idents(cd8t_apara_urbc_l1)<-"urbc"
#Idents(dc_apara_urbc_l1)<-"urbc"
Idents(mono_apara_urbc_l1)<-"urbc"
Idents(nk_apara_urbc_l1)<-"urbc"
Idents(otherT_apara_urbc_l1)<-"urbc"
Idents(other_apara_urbc_l1)<-"urbc"


#apara_irbc
Idents(b_apara_irbc_l1)<-"irbc"
Idents(cd4t_apara_irbc_l1)<-"irbc"
Idents(cd8t_apara_irbc_l1)<-"irbc"
#Idents(c_apara_irbc_l1)<-"irbc"
Idents(mono_apara_irbc_l1)<-"irbc"
Idents(nk_apara_irbc_l1)<-"irbc"
Idents(otherT_apara_irbc_l1)<-"irbc"
Idents(other_apara_irbc_l1)<-"irbc"

# predicted celltype l2
#Para uRBC
#Idents(asdc_par_urbc_l2)<-"urbc"
Idents(b_int_par_urbc_l2)<-"urbc"
Idents(b_mem_par_urbc_l2)<-"urbc"
Idents(b_naiv_par_urbc_l2)<-"urbc"
Idents(pb_par_urbc_l2)<-"urbc"
Idents(cd14_mono_par_urbc_l2)<-"urbc"
#Idents(cd16_mono_par_urbc_l2)<-"urbc"
#Idents(cd4_ctl_par_urbc_l2)<-"urbc"
Idents(cd4_naiv_par_urbc_l2)<-"urbc"
#Idents(cd4_prolif_par_urbc_l2)<-"urbc"
Idents(cd4_tcm_par_urbc_l2)<-"urbc"
Idents(cd4_tem_par_urbc_l2)<-"urbc"
Idents(cd8_naiv_par_urbc_l2)<-"urbc"
#Idents(cd8_prolif_par_urbc_l2)<-"urbc"
Idents(cd8_tcm_par_urbc_l2)<-"urbc"
Idents(cd8_tem_par_urbc_l2)<-"urbc"
Idents(treg_par_urbc_l2)<-"urbc"
Idents(gdt_par_urbc_l2)<-"urbc"
Idents(mait_par_urbc_l2)<-"urbc"
Idents(ilc_par_urbc_l2)<-"urbc"
Idents(pdc_par_urbc_l2)<-"urbc"
#Idents(cdc2_par_urbc_l2)<-"urbc"
Idents(nk_par_urbc_l2)<-"urbc"
Idents(nk_cd56br_par_urbc_l2)<-"urbc"
Idents(hspc_par_urbc_l2)<-"urbc"
Idents(dnt_par_urbc_l2)<-"urbc"
#Idents(nkprolif_par_urbc_l2)<-"urbc"
#Idents(platelet_par_urbc_l2)<-"urbc"

#Para irbc

#Idents(asdc_par_irbc_l2)<-"irbc"
Idents(b_int_par_irbc_l2)<-"irbc"
Idents(b_mem_par_irbc_l2)<-"irbc"
Idents(b_naiv_par_irbc_l2)<-"irbc"
Idents(pb_par_irbc_l2)<-"irbc"
Idents(cd14_mono_par_irbc_l2)<-"irbc"
#Idents(cd16_mono_par_irbc_l2)<-"irbc"
#Idents(cd4_ctl_par_irbc_l2)<-"irbc"
Idents(cd4_naiv_par_irbc_l2)<-"irbc"
#Idents(cd4_prolif_par_irbc_l2)<-"irbc"
Idents(cd4_tcm_par_irbc_l2)<-"irbc"
Idents(cd4_tem_par_irbc_l2)<-"irbc"
Idents(cd8_naiv_par_irbc_l2)<-"irbc"
#Idents(cd8_prolif_par_irbc_l2)<-"irbc"
Idents(cd8_tcm_par_irbc_l2)<-"irbc"
Idents(cd8_tem_par_irbc_l2)<-"irbc"
Idents(treg_par_irbc_l2)<-"irbc"
Idents(gdt_par_irbc_l2)<-"irbc"
Idents(mait_par_irbc_l2)<-"irbc"
Idents(ilc_par_irbc_l2)<-"irbc"
Idents(pdc_par_irbc_l2)<-"irbc"
#Idents(cdc2_par_irbc_l2)<-"irbc"
Idents(nk_par_irbc_l2)<-"irbc"
Idents(nk_cd56br_par_irbc_l2)<-"irbc"
Idents(hspc_par_irbc_l2)<-"irbc"
Idents(dnt_par_irbc_l2)<-"irbc"
#Idents(nkprolif_par_irbc_l2)<-"irbc"
#Idents(platelet_par_irbc_l2)<-"irbc"

#apara urbc

#Idents(asdc_apar_urbc_l2)<-"urbc"
Idents(b_int_apar_urbc_l2)<-"urbc"
Idents(b_mem_apar_urbc_l2)<-"urbc"
Idents(b_naiv_apar_urbc_l2)<-"urbc"
#Idents(pb_apar_urbc_l2)<-"urbc"
Idents(cd14_mono_apar_urbc_l2)<-"urbc"
Idents(cd16_mono_apar_urbc_l2)<-"urbc"
Idents(cd4_ctl_apar_urbc_l2)<-"urbc"
Idents(cd4_naiv_apar_urbc_l2)<-"urbc"
#Idents(cd4_prolif_apar_urbc_l2)<-"urbc"
Idents(cd4_tcm_apar_urbc_l2)<-"urbc"
Idents(cd4_tem_apar_urbc_l2)<-"urbc"
Idents(cd8_naiv_apar_urbc_l2)<-"urbc"
#Idents(cd8_prolif_apar_urbc_l2)<-"urbc"
Idents(cd8_tcm_apar_urbc_l2)<-"urbc"
Idents(cd8_tem_apar_urbc_l2)<-"urbc"
Idents(treg_apar_urbc_l2)<-"urbc"
Idents(gdt_apar_urbc_l2)<-"urbc"
Idents(mait_apar_urbc_l2)<-"urbc"
Idents(ilc_apar_urbc_l2)<-"urbc"
#Idents(pdc_apar_urbc_l2)<-"urbc"
#Idents(cdc2_apar_urbc_l2)<-"urbc"
Idents(nk_apar_urbc_l2)<-"urbc"
Idents(nk_cd56br_apar_urbc_l2)<-"urbc"
Idents(hspc_apar_urbc_l2)<-"urbc"
Idents(dnt_apar_urbc_l2)<-"urbc"
Idents(nkprolif_apar_urbc_l2)<-"urbc"
Idents(platelet_apar_urbc_l2)<-"urbc"

#apara irbc

#Idents(asdc_apar_irbc_l2)<-"irbc"
Idents(b_int_apar_irbc_l2)<-"irbc"
Idents(b_mem_apar_irbc_l2)<-"irbc"
Idents(b_naiv_apar_irbc_l2)<-"irbc"
#Idents(pb_apar_irbc_l2)<-"irbc"
Idents(cd14_mono_apar_irbc_l2)<-"irbc"
Idents(cd16_mono_apar_irbc_l2)<-"irbc"
Idents(cd4_ctl_apar_irbc_l2)<-"irbc"
Idents(cd4_naiv_apar_irbc_l2)<-"irbc"
#Idents(cd4_prolif_apar_irbc_l2)<-"irbc"
Idents(cd4_tcm_apar_irbc_l2)<-"irbc"
Idents(cd4_tem_apar_irbc_l2)<-"irbc"
Idents(cd8_naiv_apar_irbc_l2)<-"irbc"
#Idents(cd8_prolif_apar_irbc_l2)<-"irbc"
Idents(cd8_tcm_apar_irbc_l2)<-"irbc"
Idents(cd8_tem_apar_irbc_l2)<-"irbc"
Idents(treg_apar_irbc_l2)<-"irbc"
Idents(gdt_apar_irbc_l2)<-"irbc"
Idents(mait_apar_irbc_l2)<-"irbc"
Idents(ilc_apar_irbc_l2)<-"irbc"
#Idents(pdc_apar_irbc_l2)<-"irbc"
#Idents(cdc2_apar_irbc_l2)<-"irbc"
Idents(nk_apar_irbc_l2)<-"irbc"
Idents(nk_cd56br_apar_irbc_l2)<-"irbc"
Idents(hspc_apar_irbc_l2)<-"irbc"
Idents(dnt_apar_irbc_l2)<-"irbc"
Idents(nkprolif_apar_irbc_l2)<-"irbc"
Idents(platelet_apar_irbc_l2)<-"irbc"

#Merge uRBC and iRBC 
#Parasitemic
# Predicted celltype l1
bcell_para_merged_l1<-merge(b_para_urbc_l1, b_para_irbc_l1, add.cell.ids=c("urbc", "irbc"))
cd4t_para_merged_l1<-merge(cd4t_para_urbc_l1, cd4t_para_irbc_l1, add.cell.ids=c("urbc", "irbc"))
cd8t_para_merged_l1<-merge(cd8t_para_urbc_l1, cd8t_para_irbc_l1, add.cell.ids=c("urbc", "irbc"))
#dc_para_merged_l1<-merge(dc_para_urbc_l1, dc_para_irbc_l1, add.cell.ids=c("urbc", "irbc")) # throws error even though t
mono_para_merged_l1<-merge(mono_para_urbc_l1, mono_para_irbc_l1, add.cell.ids=c("urbc", "irbc"))
nk_para_merged_l1<-merge(nk_para_urbc_l1, nk_para_irbc_l1, add.cell.ids=c("urbc", "irbc"))
otherT_para_merged_l1<-merge(otherT_para_urbc_l1, otherT_para_irbc_l1, add.cell.ids=c("urbc", "irbc"))
other_para_merged_l1<-merge(other_para_urbc_l1, other_para_irbc_l1, add.cell.ids=c("urbc", "irbc"))

#Aparasitemic
# Predicted celltype l1
bcell_apara_merged_l1<-merge(b_apara_urbc_l1, b_apara_irbc_l1, add.cell.ids=c("urbc", "irbc"))
cd4t_apara_merged_l1<-merge(cd4t_apara_urbc_l1, cd4t_apara_irbc_l1, add.cell.ids=c("urbc", "irbc"))
cd8t_apara_merged_l1<-merge(cd8t_apara_urbc_l1, cd8t_apara_irbc_l1, add.cell.ids=c("urbc", "irbc"))
#dc_apara_merged_l1<-merge(dc_apara_urbc_l1, dc_apara_irbc_l1, add.cell.ids=c("urbc", "irbc"))
#for mono, remove ATAC assay from mono_apara_irbc_l1, which is not present in mono_apara_urbc_l1
mono_apara_irbc_l1[["ATAC"]]<-NULL
mono_apara_merged_l1<-merge(mono_apara_urbc_l1, mono_apara_irbc_l1, add.cell.ids=c("urbc", "irbc"))
nk_apara_merged_l1<-merge(nk_apara_urbc_l1, nk_apara_irbc_l1, add.cell.ids=c("urbc", "irbc"))
otherT_apara_merged_l1<-merge(otherT_apara_urbc_l1, otherT_apara_irbc_l1, add.cell.ids=c("urbc", "irbc"))
#other_apara_merged_l1<-merge(other_apara_urbc_l1, other_apara_irbc_l1, add.cell.ids=c("urbc", "irbc")) # have same

#Parasitemic
# Predicted celltype l2
#asdc_para_merged_l2<-merge(asdc_par_urbc_l2, asdc_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
b_int_para_merged_l2<-merge(b_int_par_urbc_l2, b_int_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#for bmem, remove ATAC assay from b_mem_par_irbc_l2, which is not present in mb_mem_par_urbc_l2
b_mem_par_irbc_l2[["ATAC"]]<-NULL
b_mem_para_merged_l2<-merge(b_mem_par_urbc_l2, b_mem_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
b_naiv_para_merged_l2<-merge(b_naiv_par_urbc_l2, b_naiv_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
pb_para_merged_l2<-merge(pb_par_urbc_l2, pb_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd14_mono_para_merged_l2<-merge(cd14_mono_par_urbc_l2, cd14_mono_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cd16_mono_para_merged_l2<-merge(cd16_mono_par_urbc_l2, cd16_mono_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cd4_ctl_para_merged_l2<-merge(cd4_ctl_par_urbc_l2, cd4_ctl_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd4_naiv_para_merged_l2<-merge(cd4_naiv_par_urbc_l2, cd4_naiv_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cd4_prolif_para_merged_l2<-merge(cd4_prolif_par_urbc_l2, cd4_prolif_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd4_tcm_para_merged_l2<-merge(cd4_tcm_par_urbc_l2, cd4_tcm_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cd4_tem_para_merged_l2<-merge(cd4_tem_par_urbc_l2, cd4_tem_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd8_naiv_para_merged_l2<-merge(cd8_naiv_par_urbc_l2, cd8_naiv_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd8_tcm_para_merged_l2<-merge(cd8_tcm_par_urbc_l2, cd8_tcm_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd8_tem_para_merged_l2<-merge(cd8_tem_par_urbc_l2, cd8_tem_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
treg_para_merged_l2<-merge(treg_par_urbc_l2, treg_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
gdt_para_merged_l2<-merge(gdt_par_urbc_l2, gdt_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#mait_para_merged_l2<-merge(mait_par_urbc_l2, mait_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
ilc_para_merged_l2<-merge(ilc_par_urbc_l2, ilc_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#pdc_para_merged_l2<-merge(pdc_par_urbc_l2, pdc_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cdc2_para_merged_l2<-merge(cdc2_par_urbc_l2, cdc2_par_irbc_l2, add.cell.ids=c("urbc", "irbc")) 
nk_para_merged_l2<-merge(nk_par_urbc_l2, nk_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#nk_cd56br_para_merged_l2<-merge(nk_cd56br_par_urbc_l2, nk_cd56br_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#hspc_para_merged_l2<-merge(hspc_par_urbc_l2, hspc_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
dnt_para_merged_l2<-merge(dnt_par_urbc_l2, dnt_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#nkprolif_para_merged_l2<-merge(nkprolif_par_urbc_l2, nkprolif_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#platelet_para_merged_l2<-merge(platelet_par_urbc_l2, platelet_par_irbc_l2, add.cell.ids=c("urbc", "irbc"))

#Aparasitemic
# Predicted celltype l2
#asdc_apara_merged_l2<-merge(asdc_apar_urbc_l2, asdc_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
b_int_apara_merged_l2<-merge(b_int_apar_urbc_l2, b_int_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
b_mem_apara_merged_l2<-merge(b_mem_apar_urbc_l2, b_mem_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
b_naiv_apara_merged_l2<-merge(b_naiv_apar_urbc_l2, b_naiv_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#pb_apara_merged_l2<-merge(pb_apar_urbc_l2, pb_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#for cd14mono, remove ATAC assay from cd14_mono_apar_irbc_l2, which is not present in cd14_mono_apar_urbc_l2
cd14_mono_apar_irbc_l2[["ATAC"]]<-NULL
cd14_mono_apara_merged_l2<-merge(cd14_mono_apar_urbc_l2, cd14_mono_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd16_mono_apara_merged_l2<-merge(cd16_mono_apar_urbc_l2, cd16_mono_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cd4_ctl_apara_merged_l2<-merge(cd4_ctl_apar_urbc_l2, cd4_ctl_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd4_naiv_apara_merged_l2<-merge(cd4_naiv_apar_urbc_l2, cd4_naiv_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cd4_prolif_apara_merged_l2<-merge(cd4_prolif_apar_urbc_l2, cd4_prolif_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cd4_tcm_apara_merged_l2<-merge(cd4_tcm_apar_urbc_l2, cd4_tcm_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd4_tem_apara_merged_l2<-merge(cd4_tem_apar_urbc_l2, cd4_tem_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd8_naiv_apara_merged_l2<-merge(cd8_naiv_apar_urbc_l2, cd8_naiv_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd8_tcm_apara_merged_l2<-merge(cd8_tcm_apar_urbc_l2, cd8_tcm_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
cd8_tem_apara_merged_l2<-merge(cd8_tem_apar_urbc_l2, cd8_tem_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
treg_apara_merged_l2<-merge(treg_apar_urbc_l2, treg_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
gdt_apara_merged_l2<-merge(gdt_apar_urbc_l2, gdt_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#mait_apara_merged_l2<-merge(mait_apar_urbc_l2, mait_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#ilc_apara_merged_l2<-merge(ilc_apar_urbc_l2, ilc_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#pdc_apara_merged_l2<-merge(pdc_apar_urbc_l2, pdc_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#cdc2_apara_merged_l2<-merge(cdc2_apar_urbc_l2, cdc2_apar_irbc_l2, add.cell.ids=c("urbc", "irbc")) 
nk_apara_merged_l2<-merge(nk_apar_urbc_l2, nk_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
nk_cd56br_apara_merged_l2<-merge(nk_cd56br_apar_urbc_l2, nk_cd56br_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
#hspc_apara_merged_l2<-merge(hspc_apar_urbc_l2, hspc_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
dnt_apara_merged_l2<-merge(dnt_apar_urbc_l2, dnt_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
nkprolif_apara_merged_l2<-merge(nkprolif_apar_urbc_l2, nkprolif_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
platelet_apara_merged_l2<-merge(platelet_apar_urbc_l2, platelet_apar_irbc_l2, add.cell.ids=c("urbc", "irbc"))
```

```{r}
# To make the merged objects comparable, run PrepSCTFindMarkers on each object
# Predicted celltype l1
#Para
bcell_para_merged_l1<-PrepSCTFindMarkers(bcell_para_merged_l1)
cd4t_para_merged_l1<-PrepSCTFindMarkers(cd4t_para_merged_l1)
cd8t_para_merged_l1<-PrepSCTFindMarkers(cd8t_para_merged_l1)
#dc_para_merged_l1<-PrepSCTFindMarkers(dc_para_merged_l1)
mono_para_merged_l1<-PrepSCTFindMarkers(mono_para_merged_l1)
nk_para_merged_l1<-PrepSCTFindMarkers(nk_para_merged_l1)
otherT_para_merged_l1<-PrepSCTFindMarkers(otherT_para_merged_l1)
#other_para_merged_l1<-PrepSCTFindMarkers(other_para_merged_l1)

#Apara
# Predicted celltype l1
bcell_apara_merged_l1<-PrepSCTFindMarkers(bcell_apara_merged_l1)
cd4t_apara_merged_l1<-PrepSCTFindMarkers(cd4t_apara_merged_l1)
cd8t_apara_merged_l1<-PrepSCTFindMarkers(cd8t_apara_merged_l1)
#dc_apara_merged_l1<-PrepSCTFindMarkers(dc_apara_merged_l1)
mono_apara_merged_l1<-PrepSCTFindMarkers(mono_apara_merged_l1)
nk_apara_merged_l1<-PrepSCTFindMarkers(nk_apara_merged_l1)
otherT_apara_merged_l1<-PrepSCTFindMarkers(otherT_apara_merged_l1)
#other_apara_merged_l1<-PrepSCTFindMarkers(other_apara_merged_l1)

# Predicted celltype l2
#Para
#asdc_para_merged_l2<-PrepSCTFindMarkers(asdc_para_merged_l2)
b_int_para_merged_l2<-PrepSCTFindMarkers(b_int_para_merged_l2)
b_mem_para_merged_l2<-PrepSCTFindMarkers(b_mem_para_merged_l2)
b_naiv_para_merged_l2<-PrepSCTFindMarkers(b_naiv_para_merged_l2)
#pb_para_merged_l2<-PrepSCTFindMarkers(pb_para_merged_l2)
cd14_mono_para_merged_l2<-PrepSCTFindMarkers(cd14_mono_para_merged_l2)
#cd16_mono_para_merged_l2<-PrepSCTFindMarkers(cd16_mono_para_merged_l2)
#cd4_ctl_para_merged_l2<-PrepSCTFindMarkers(cd4_ctl_para_merged_l2)
cd4_naiv_para_merged_l2<-PrepSCTFindMarkers(cd4_naiv_para_merged_l2)
#cd4_prolif_para_merged_l2<-PrepSCTFindMarkers(cd4_prolif_para_merged_l2)
#cd4_tcm_para_merged_l2<-PrepSCTFindMarkers(cd4_tcm_para_merged_l2)
#cd4_tem_para_merged_l2<-PrepSCTFindMarkers(cd4_tem_para_merged_l2)
cd8_naiv_para_merged_l2<-PrepSCTFindMarkers(cd8_naiv_para_merged_l2)
cd8_tcm_para_merged_l2<-PrepSCTFindMarkers(cd8_tcm_para_merged_l2)
cd8_tem_para_merged_l2<-PrepSCTFindMarkers(cd8_tem_para_merged_l2)
treg_para_merged_l2<-PrepSCTFindMarkers(treg_para_merged_l2)
gdt_para_merged_l2<-PrepSCTFindMarkers(gdt_para_merged_l2)
#mait_para_merged_l2<-PrepSCTFindMarkers(mait_para_merged_l2)
#ilc_para_merged_l2<-PrepSCTFindMarkers(ilc_para_merged_l2)
#pdc_para_merged_l2<-PrepSCTFindMarkers(pdc_para_merged_l2)
#cdc2_para_merged_l2<-PrepSCTFindMarkers(cdc2_para_merged_l2)
nk_para_merged_l2<-PrepSCTFindMarkers(nk_para_merged_l2)
#nk_cd56br_para_merged_l2<-PrepSCTFindMarkers(nk_cd56br_para_merged_l2)
#hspc_para_merged_l2<-PrepSCTFindMarkers(hspc_para_merged_l2)
dnt_para_merged_l2<-PrepSCTFindMarkers(dnt_para_merged_l2)
#nkprolif_para_merged_l2<-PrepSCTFindMarkers(nkprolif_para_merged_l2)
#platelet_para_merged_l2<-PrepSCTFindMarkers(platelet_para_merged_l2)

#Apara
#asdc_apara_merged_l2<-PrepSCTFindMarkers(asdc_apara_merged_l2)
b_int_apara_merged_l2<-PrepSCTFindMarkers(b_int_apara_merged_l2)
b_mem_apara_merged_l2<-PrepSCTFindMarkers(b_mem_apara_merged_l2)
b_naiv_apara_merged_l2<-PrepSCTFindMarkers(b_naiv_apara_merged_l2)
#pb_apara_merged_l2<-PrepSCTFindMarkers(pb_apara_merged_l2)
cd14_mono_apara_merged_l2<-PrepSCTFindMarkers(cd14_mono_apara_merged_l2)
#cd16_mono_apara_merged_l2<-PrepSCTFindMarkers(cd16_mono_apara_merged_l2)
#cd4_ctl_apara_merged_l2<-PrepSCTFindMarkers(cd4_ctl_apara_merged_l2)
cd4_naiv_apara_merged_l2<-PrepSCTFindMarkers(cd4_naiv_apara_merged_l2)
#cd4_prolif_apara_merged_l2<-PrepSCTFindMarkers(cd4_prolif_apara_merged_l2)
#cd4_tcm_apara_merged_l2<-PrepSCTFindMarkers(cd4_tcm_apara_merged_l2)
#cd4_tem_apara_merged_l2<-PrepSCTFindMarkers(cd4_tem_apara_merged_l2)
cd8_naiv_apara_merged_l2<-PrepSCTFindMarkers(cd8_naiv_apara_merged_l2)
cd8_tcm_apara_merged_l2<-PrepSCTFindMarkers(cd8_tcm_apara_merged_l2)
cd8_tem_apara_merged_l2<-PrepSCTFindMarkers(cd8_tem_apara_merged_l2)
treg_apara_merged_l2<-PrepSCTFindMarkers(treg_apara_merged_l2)
gdt_apara_merged_l2<-PrepSCTFindMarkers(gdt_apara_merged_l2)
#mait_apara_merged_l2<-PrepSCTFindMarkers(mait_apara_merged_l2)
#ilc_apara_merged_l2<-PrepSCTFindMarkers(ilc_apara_merged_l2)
#pdc_apara_merged_l2<-PrepSCTFindMarkers(pdc_apara_merged_l2)
#cdc2_apara_merged_l2<-PrepSCTFindMarkers(cdc2_apara_merged_l2)
nk_apara_merged_l2<-PrepSCTFindMarkers(nk_apara_merged_l2)
#nk_cd56br_apara_merged_l2<-PrepSCTFindMarkers(nk_cd56br_apara_merged_l2)
#hspc_apara_merged_l2<-PrepSCTFindMarkers(hspc_apara_merged_l2)
dnt_apara_merged_l2<-PrepSCTFindMarkers(dnt_apara_merged_l2)
#nkprolif_apara_merged_l2<-PrepSCTFindMarkers(nkprolif_apara_merged_l2)
#platelet_apara_merged_l2<-PrepSCTFindMarkers(platelet_apara_merged_l2)
```
#Use traditional FindMarkers, as cells are not enough to add randome effect for subject (<50 in many cases)

# Use modified funcitons to do DE analysis
```{r}
# And now we can call our FindMarkers with random effects model for sample_ID
# Don't forget to include ebayes = F
#Para
bcell_para_deglist_l1<- FindMarkers(bcell_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd4t_para_deglist_l1<- FindMarkers(cd4t_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8t_para_deglist_l1<- FindMarkers(cd8t_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
dc_para_deglist_l1<- FindMarkers(dc_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
mono_para_deglist_l1<- FindMarkers(mono_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
nk_para_deglist_l1<- FindMarkers(nk_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
otherT_para_deglist_l1<- FindMarkers(otherT_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
other_para_deglist_l1<- FindMarkers(other_para_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)

#Apara
bcell_apara_deglist_l1<- FindMarkers(bcell_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd4t_apara_deglist_l1<- FindMarkers(cd4t_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8t_apara_deglist_l1<- FindMarkers(cd8t_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
dc_apara_deglist_l1<- FindMarkers(dc_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
mono_apara_deglist_l1<- FindMarkers(mono_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
nk_apara_deglist_l1<- FindMarkers(nk_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
otherT_apara_deglist_l1<- FindMarkers(otherT_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
other_apara_deglist_l1<- FindMarkers(other_apara_merged_l1,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)

# Predicted celltype L2
#Para
#asdc_para_deglist_l2 <- FindMarkers(asdc_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
b_int_para_deglist_l2 <- FindMarkers(b_int_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
b_mem_para_deglist_l2 <- FindMarkers(b_mem_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
b_naiv_para_deglist_l2 <- FindMarkers(b_naiv_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#pb_para_deglist_l2 <- FindMarkers(pb_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd14_mono_para_deglist_l2 <- FindMarkers(cd14_mono_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd16_mono_para_deglist_l2 <- FindMarkers(cd16_mono_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd4_ctl_para_deglist_l2 <- FindMarkers(cd4_ctl_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
# warns on CD4 naive
cd4_naiv_para_deglist_l2 <- FindMarkers(cd4_naiv_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd4_prolif_para_deglist_l2 <- FindMarkers(cd4_prolif_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
# warns on CD4 naive
#cd4_tcm_para_deglist_l2 <- FindMarkers(cd4_tcm_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd4_tem_para_deglist_l2 <- FindMarkers(cd4_tem_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8_naiv_para_deglist_l2 <- FindMarkers(cd8_naiv_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8_tcm_para_deglist_l2 <- FindMarkers(cd8_tcm_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8_tem_para_deglist_l2 <- FindMarkers(cd8_tem_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
treg_para_deglist_l2 <- FindMarkers(cd8_tem_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
gdt_para_deglist_l2 <- FindMarkers(gdt_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#mait_para_deglist_l2 <- FindMarkers(mait_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#ilc_para_deglist_l2 <- FindMarkers(ilc_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#error
#pdc_para_deglist_l2 <- FindMarkers(pdc_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cdc2_para_deglist_l2 <- FindMarkers(cdc2_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
nk_para_deglist_l2 <- FindMarkers(nk_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#nk_cd56br_para_deglist_l2 <- FindMarkers(nk_cd56br_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#hspc_para_deglist_l2 <- FindMarkers(hspc_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
dnt_para_deglist_l2 <- FindMarkers(dnt_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#nkprolif_para_deglist_l2 <- FindMarkers(nkprolif_para_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)

#apara
#asdc_apara_deglist_l2 <- FindMarkers(asdc_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
b_int_apara_deglist_l2 <- FindMarkers(b_int_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
b_mem_apara_deglist_l2 <- FindMarkers(b_mem_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
b_naiv_apara_deglist_l2 <- FindMarkers(b_naiv_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#pb_apara_deglist_l2 <- FindMarkers(pb_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd14_mono_apara_deglist_l2 <- FindMarkers(cd14_mono_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd16_mono_apara_deglist_l2 <- FindMarkers(cd16_mono_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd4_ctl_apara_deglist_l2 <- FindMarkers(cd4_ctl_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
# warns on CD4 naive
cd4_naiv_apara_deglist_l2 <- FindMarkers(cd4_naiv_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd4_prolif_apara_deglist_l2 <- FindMarkers(cd4_prolif_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
# warns on CD4 naive
#cd4_tcm_apara_deglist_l2 <- FindMarkers(cd4_tcm_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cd4_tem_apara_deglist_l2 <- FindMarkers(cd4_tem_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8_naiv_apara_deglist_l2 <- FindMarkers(cd8_naiv_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8_tcm_apara_deglist_l2 <- FindMarkers(cd8_tcm_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
cd8_tem_apara_deglist_l2 <- FindMarkers(cd8_tem_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
treg_apara_deglist_l2 <- FindMarkers(cd8_tem_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
gdt_apara_deglist_l2 <- FindMarkers(gdt_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#mait_apara_deglist_l2 <- FindMarkers(mait_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#ilc_apara_deglist_l2 <- FindMarkers(ilc_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#error
#pdc_apara_deglist_l2 <- FindMarkers(pdc_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#cdc2_apara_deglist_l2 <- FindMarkers(cdc2_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
nk_apara_deglist_l2 <- FindMarkers(nk_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#nk_cd56br_apara_deglist_l2 <- FindMarkers(nk_cd56br_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#hspc_apara_deglist_l2 <- FindMarkers(hspc_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
dnt_apara_deglist_l2 <- FindMarkers(dnt_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
#nkprolif_apara_deglist_l2 <- FindMarkers(nkprolif_apara_merged_l2,ident.1 = "irbc", ident.2 = "urbc",only.pos = F, test.use = "MAST", assay = "RNA",  logfc.threshold = 0.05,  min.pct = 0.1)
```

```{r}
# Concatenate all dataframes of DEGlists together
allcelltypes_exvivo_deglists <- mget(ls(pattern = "*deglist"))

# Save
saveRDS(allcelltypes_exvivo_deglists, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/In vitro/R21SterileImmunity_invitro_CITE_Multiome_allcelltypes_DEGList_iRBC_vs_uRBC_forapara_and_para_test_MAST_with_no_adj_for_sample_05_01_2023.rds")
```

#GSEA on all celltypes
```{r}
# Optional: Import DEGlists from Google Drive
temp <- tempfile(fileext = ".rds")
dl <- drive_download(
  as_id(""), path = temp, overwrite = TRUE)
deglists <- readRDS(file = dl$local_path)

# Deconvolute into individual celltypes' DEGlists
lapply(names(deglists), function(x) assign(x, deglists[[x]], envir = .GlobalEnv))
```

#predicted celltype L1
```{r GSE on predicted celltype L1}
#Parasitemic
# bcell
bcell_para_deglist_l1<-bcell_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

bcell_para_rank_l1<-bcell_para_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l1_bcell<-NamedGeneRankList2GseaTable(rankedgenes = bcell_para_rank_l1,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_para_l1_bcell<- gsea_para_l1_bcell%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
# cd4t cell
cd4t_para_deglist_l1<-cd4t_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

cd4t_para_rank_l1<-cd4t_para_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>% ########### remove infinite values in rank that seem to throw an error in next step
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l1_cd4t<-NamedGeneRankList2GseaTable(rankedgenes = cd4t_para_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_para_l1_cd4t<- gsea_para_l1_cd4t%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# cd8t cell
cd8t_para_deglist_l1<-cd8t_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

cd8t_para_rank_l1<-cd8t_para_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l1_cd8t<-NamedGeneRankList2GseaTable(rankedgenes = cd8t_para_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_para_l1_cd8t<- gsea_para_l1_cd8t%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# monocyte
mono_para_deglist_l1<-mono_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

mono_para_rank_l1<-mono_para_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l1_mono<-NamedGeneRankList2GseaTable(rankedgenes = mono_para_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_para_l1_mono<- gsea_para_l1_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# dendritic cell
#dc_para_deglist_l1<-dc_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

#dc_para_rank_l1<-dc_para_deglist_l1%>%
 # mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
#  dplyr::select(GeneSymbol,rankmetric) %>%
#  na.omit() %>%
#  distinct() %>%
#  group_by(GeneSymbol) %>%
#  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
#  dplyr::filter_all(all_vars(is.finite(.)))%>%
#  arrange(desc(rankmetric)) %>%
#  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

#gsea_para_l1_dc<-NamedGeneRankList2GseaTable(rankedgenes = dc_para_rank_l1,
 #                                       geneset = "all",
 #                                       output_directory = tempdir(),
 #                                       filename_prefix = "GSEA",
#                                        sampleSize = 101,
#                                        minSize = 20,
 #                                       maxSize = Inf,
 #                                       scoreType = "std")

#gsea_para_l1_dc<- gsea_para_l1_dc%>%
#  as_tibble() %>%
 # arrange(desc(NES)) %>% 
 # dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
#  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# nk cell
nk_para_deglist_l1<-nk_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

nk_para_rank_l1<-nk_para_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l1_nk<-NamedGeneRankList2GseaTable(rankedgenes = nk_para_rank_l1,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_para_l1_nk<- gsea_para_l1_nk%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# Other T cell
otherT_para_deglist_l1<-otherT_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

otherT_para_rank_l1<-otherT_para_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l1_otherT<-NamedGeneRankList2GseaTable(rankedgenes = otherT_para_rank_l1,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_para_l1_otherT<- gsea_para_l1_otherT%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# Other
#other_para_deglist_l1<-other_para_deglist_l1%>%rownames_to_column(., "GeneSymbol")

#other_para_rank_l1<-other_para_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

#gsea_para_l1_other<-NamedGeneRankList2GseaTable(rankedgenes = other_para_rank_l1,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

#gsea_para_l1_other<- gsea_para_l1_other%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()
  
  
#aparasitemic
# bcell
bcell_apara_deglist_l1<-bcell_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

bcell_apara_rank_l1<-bcell_apara_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l1_bcell<-NamedGeneRankList2GseaTable(rankedgenes = bcell_apara_rank_l1,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_apara_l1_bcell<- gsea_apara_l1_bcell%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
# cd4t cell
cd4t_apara_deglist_l1<-cd4t_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

cd4t_apara_rank_l1<-cd4t_apara_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>% ########### remove infinite values in rank that seem to throw an error in next step
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l1_cd4t<-NamedGeneRankList2GseaTable(rankedgenes = cd4t_apara_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_apara_l1_cd4t<- gsea_apara_l1_cd4t%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# cd8t cell
cd8t_apara_deglist_l1<-cd8t_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

cd8t_apara_rank_l1<-cd8t_apara_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l1_cd8t<-NamedGeneRankList2GseaTable(rankedgenes = cd8t_apara_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_apara_l1_cd8t<- gsea_apara_l1_cd8t%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# monocyte
mono_apara_deglist_l1<-mono_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

mono_apara_rank_l1<-mono_apara_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l1_mono<-NamedGeneRankList2GseaTable(rankedgenes = mono_apara_rank_l1,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_apara_l1_mono<- gsea_apara_l1_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# dendritic cell
#dc_apara_deglist_l1<-dc_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

#dc_apara_rank_l1<-dc_apara_deglist_l1%>%
 # mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
#  dplyr::select(GeneSymbol,rankmetric) %>%
#  na.omit() %>%
#  distinct() %>%
#  group_by(GeneSymbol) %>%
#  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
#  dplyr::filter_all(all_vars(is.finite(.)))%>%
#  arrange(desc(rankmetric)) %>%
#  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

#gsea_apara_l1_dc<-NamedGeneRankList2GseaTable(rankedgenes = dc_apara_rank_l1,
 #                                       geneset = "all",
 #                                       output_directory = tempdir(),
 #                                       filename_prefix = "GSEA",
#                                        sampleSize = 101,
#                                        minSize = 20,
 #                                       maxSize = Inf,
 #                                       scoreType = "std")

#gsea_apara_l1_dc<- gsea_apara_l1_dc%>%
#  as_tibble() %>%
 # arrange(desc(NES)) %>% 
 # dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
#  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# nk cell
nk_apara_deglist_l1<-nk_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

nk_apara_rank_l1<-nk_apara_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l1_nk<-NamedGeneRankList2GseaTable(rankedgenes = nk_apara_rank_l1,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_apara_l1_nk<- gsea_apara_l1_nk%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# Other T cell
otherT_apara_deglist_l1<-otherT_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

otherT_apara_rank_l1<-otherT_apara_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l1_otherT<-NamedGeneRankList2GseaTable(rankedgenes = otherT_apara_rank_l1,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_apara_l1_otherT<- gsea_apara_l1_otherT%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()


##########################################################################################################################################
# Other
#other_apara_deglist_l1<-other_apara_deglist_l1%>%rownames_to_column(., "GeneSymbol")

#other_apara_rank_l1<-other_apara_deglist_l1%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

#gsea_apara_l1_other<-NamedGeneRankList2GseaTable(rankedgenes = other_apara_rank_l1,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

#gsea_apara_l1_other<- gsea_apara_l1_other%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()
```


#Predicted celltype l2
# Run GSEA on each celltype in predictedceltlype L2
```{r GSEA of Predicted celltypes L2}
#Parasitemic
#ASDC - no ASDcs in aparasitemic kids
#asdc_para_deglist_l2<-asdc_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

#asdc_para_rank_l2<-asdc_para_deglist_l2%>%
#  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
#  dplyr::select(GeneSymbol,rankmetric) %>%
#  na.omit() %>%
#  distinct() %>%
#  group_by(GeneSymbol) %>%
#  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
#  arrange(desc(rankmetric)) %>%
#  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

#gsea_para_l2_asdc<-NamedGeneRankList2GseaTable(rankedgenes = asdc_para_rank_l2,
#                                       geneset = "all",
#                                       output_directory = tempdir(),
#                                      filename_prefix = "GSEA",
#                                       sampleSize = 101,
#                                       minSize = 20,
#                                       maxSize = Inf,
#                                       scoreType = "std")

#gsea_para_l2_asdc<- gsea_para_l2_asdc%>%
#  as_tibble() %>%
#  arrange(desc(NES)) %>% 
#  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
#  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
#  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
#  arrange(padj)%>%
#filter(padj < 0.20) %>%
#  filter(!grepl("TBA", pathway)) %>%
#  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
#  mutate(neglogpadj = -log10(padj)) %>%
#  mutate(pathway = gsub("gd", "", pathway)) %>%
#  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
#  mutate(pathway = gsub("Vg", "V", pathway)) %>%
#  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
#  mutate(pathway = gsub("_", " ", pathway)) %>%
#  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
#  group_by(module_type) %>%
#  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
#  ungroup() %>%
#  filter(!grepl("TBD", pathway)) %>%
#  arrange(desc(neglogpadj)) %>%
#  droplevels()

##########################################################################################################################################
#b_int
b_int_para_deglist_l2<-b_int_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_int_para_rank_l2<-b_int_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_b_int<-NamedGeneRankList2GseaTable(rankedgenes = b_int_para_rank_l2,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_para_l2_b_int<- gsea_para_l2_b_int%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#b_mem
b_mem_para_deglist_l2<-b_mem_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_mem_para_rank_l2<-b_mem_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_b_mem<-NamedGeneRankList2GseaTable(rankedgenes = b_mem_para_rank_l2,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_para_l2_b_mem<- gsea_para_l2_b_mem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#b_naiv
b_naiv_para_deglist_l2<-b_naiv_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_naiv_para_rank_l2<-b_naiv_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_b_naiv<-NamedGeneRankList2GseaTable(rankedgenes = b_naiv_para_rank_l2,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_para_l2_b_naiv<- gsea_para_l2_b_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#pb
pb_para_deglist_l2<-pb_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

pb_para_rank_l2<-pb_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_pb<-NamedGeneRankList2GseaTable(rankedgenes = pb_para_rank_l2,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_para_l2_pb<- gsea_para_l2_pb%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd14_mono
cd14_mono_para_deglist_l2<-cd14_mono_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd14_mono_para_rank_l2<-cd14_mono_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd14_mono<-NamedGeneRankList2GseaTable(rankedgenes = cd14_mono_para_rank_l2,
                                               geneset = "all",
                                               output_directory = tempdir(),
                                               filename_prefix = "GSEA",
                                               sampleSize = 101,
                                               minSize = 20,
                                               maxSize = Inf,
                                               scoreType = "std")

gsea_para_l2_cd14_mono<- gsea_para_l2_cd14_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
 # mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd16_mono
cd16_mono_para_deglist_l2<-cd16_mono_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd16_mono_para_rank_l2<-cd16_mono_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd16_mono<-NamedGeneRankList2GseaTable(rankedgenes = cd16_mono_para_rank_l2,
                                               geneset = "all",
                                               output_directory = tempdir(),
                                               filename_prefix = "GSEA",
                                               sampleSize = 101,
                                               minSize = 20,
                                               maxSize = Inf,
                                               scoreType = "std")

gsea_para_l2_cd16_mono<- gsea_para_l2_cd16_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_ctl
cd4_ctl_para_deglist_l2<-cd4_ctl_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_ctl_para_rank_l2<-cd4_ctl_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd4_ctl<-NamedGeneRankList2GseaTable(rankedgenes = cd4_ctl_para_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_para_l2_cd4_ctl<- gsea_para_l2_cd4_ctl%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_naiv
cd4_naiv_para_deglist_l2<-cd4_naiv_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_naiv_para_rank_l2<-cd4_naiv_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd4_naiv<-NamedGeneRankList2GseaTable(rankedgenes = cd4_naiv_para_rank_l2,
                                              geneset = "all",
                                              output_directory = tempdir(),
                                              filename_prefix = "GSEA",
                                              sampleSize = 101,
                                              minSize = 20,
                                              maxSize = Inf,
                                              scoreType = "std")

gsea_para_l2_cd4_naiv<- gsea_para_l2_cd4_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_prolif
cd4_prolif_para_deglist_l2<-cd4_prolif_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_prolif_para_rank_l2<-cd4_prolif_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd4_prolif<-NamedGeneRankList2GseaTable(rankedgenes = cd4_prolif_para_rank_l2,
                                                geneset = "all",
                                                output_directory = tempdir(),
                                                filename_prefix = "GSEA",
                                                sampleSize = 101,
                                                minSize = 20,
                                                maxSize = Inf,
                                                scoreType = "std")

gsea_para_l2_cd4_prolif<- gsea_para_l2_cd4_prolif%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_tcm
cd4_tcm_para_deglist_l2<-cd4_tcm_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_tcm_para_rank_l2<-cd4_tcm_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd4_tcm<-NamedGeneRankList2GseaTable(rankedgenes = cd4_tcm_para_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_para_l2_cd4_tcm<- gsea_para_l2_cd4_tcm%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_tem
cd4_tem_para_deglist_l2<-cd4_tem_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_tem_para_rank_l2<-cd4_tem_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd4_tem<-NamedGeneRankList2GseaTable(rankedgenes = cd4_tem_para_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_para_l2_cd4_tem<- gsea_para_l2_cd4_tem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_naiv
cd8_naiv_para_deglist_l2<-cd8_naiv_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_naiv_para_rank_l2<-cd8_naiv_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd8_naiv<-NamedGeneRankList2GseaTable(rankedgenes = cd8_naiv_para_rank_l2,
                                              geneset = "all",
                                              output_directory = tempdir(),
                                              filename_prefix = "GSEA",
                                              sampleSize = 101,
                                              minSize = 20,
                                              maxSize = Inf,
                                              scoreType = "std")

gsea_para_l2_cd8_naiv<- gsea_para_l2_cd8_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_tcm
cd8_tcm_para_deglist_l2<-cd8_tcm_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_tcm_para_rank_l2<-cd8_tcm_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd8_tcm<-NamedGeneRankList2GseaTable(rankedgenes = cd8_tcm_para_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_para_l2_cd8_tcm<- gsea_para_l2_cd8_tcm%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_tem
cd8_tem_para_deglist_l2<-cd8_tem_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_tem_para_rank_l2<-cd8_tem_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cd8_tem<-NamedGeneRankList2GseaTable(rankedgenes = cd8_tem_para_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_para_l2_cd8_tem<- gsea_para_l2_cd8_tem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#treg
treg_para_deglist_l2<-treg_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

treg_para_rank_l2<-treg_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_treg<-NamedGeneRankList2GseaTable(rankedgenes = treg_para_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_para_l2_treg<- gsea_para_l2_treg%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#gdt
gdt_para_deglist_l2<-gdt_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

gdt_para_rank_l2<-gdt_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_gdt<-NamedGeneRankList2GseaTable(rankedgenes = gdt_para_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_para_l2_gdt<- gsea_para_l2_gdt%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#mait
mait_para_deglist_l2<-mait_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

mait_para_rank_l2<-mait_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_mait<-NamedGeneRankList2GseaTable(rankedgenes = mait_para_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_para_l2_mait<- gsea_para_l2_mait%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#ilc
ilc_para_deglist_l2<-ilc_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

ilc_para_rank_l2<-ilc_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_ilc<-NamedGeneRankList2GseaTable(rankedgenes = ilc_para_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_para_l2_ilc<- gsea_para_l2_ilc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#pdc
pdc_para_deglist_l2<-pdc_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

pdc_para_rank_l2<-pdc_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_pdc<-NamedGeneRankList2GseaTable(rankedgenes = pdc_para_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_para_l2_pdc<- gsea_para_l2_pdc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cdc2
cdc2_para_deglist_l2<-cdc2_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cdc2_para_rank_l2<-cdc2_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_cdc2<-NamedGeneRankList2GseaTable(rankedgenes = cdc2_para_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_para_l2_cdc2<- gsea_para_l2_cdc2%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################

##########################################################################################################################################
#nk
nk_para_deglist_l2<-nk_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nk_para_rank_l2<-nk_para_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_nk<-NamedGeneRankList2GseaTable(rankedgenes = nk_para_rank_l2,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_para_l2_nk<- gsea_para_l2_nk%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#nk_cd56br
nk_cd56br_para_deglist_l2<-nk_cd56br_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nk_cd56br_para_rank_l2_cd56br<-nk_cd56br_para_deglist_l2%>%
  mutate(rank_cd56brmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rank_cd56brmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rank_cd56brmetric = mean(rank_cd56brmetric)) %>%
  arrange(desc(rank_cd56brmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_nk_cd56br<-NamedGeneRankList2GseaTable(rankedgenes = nk_cd56br_para_rank_l2_cd56br,
                                               geneset = "all",
                                               output_directory = tempdir(),
                                               filename_prefix = "GSEA",
                                               sampleSize = 101,
                                               minSize = 20,
                                               maxSize = Inf,
                                               scoreType = "std")

gsea_para_l2_nk_cd56br<- gsea_para_l2_nk_cd56br%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#nkprolif
nkprolif_para_deglist_l2<-nkprolif_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nkprolif_para_rank_l2prolif<-nkprolif_para_deglist_l2%>%
  mutate(rankprolifmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankprolifmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankprolifmetric = mean(rankprolifmetric)) %>%
  arrange(desc(rankprolifmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_nkprolif<-NamedGeneRankList2GseaTable(rankedgenes = nkprolif_para_rank_l2prolif,
                                              geneset = "all",
                                              output_directory = tempdir(),
                                              filename_prefix = "GSEA",
                                              sampleSize = 101,
                                              minSize = 20,
                                              maxSize = Inf,
                                              scoreType = "std")

gsea_para_l2_nkprolif<- gsea_para_l2_nkprolif%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  # mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

#Untitled 4
##########################################################################################################################################
#hspc
hspc_para_deglist_l2<-hspc_para_deglist_l2%>%rownames_to_column(., "GeneSymbol")

hspc_rahspc<-hspc_para_deglist_l2%>%
  mutate(rahspcmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rahspcmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rahspcmetric = mean(rahspcmetric)) %>%
  arrange(desc(rahspcmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_para_l2_hspc<-NamedGeneRankList2GseaTable(rankedgenes = hspc_rahspc,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_para_l2_hspc<- gsea_para_l2_hspc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  # mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

# Run GSEA on each celltype in predictedceltlype L2
```{r GSEA of Predicted celltypes L2}
#aparasitemic
#ASDC - no ASDcs in aaparasitemic kids
#asdc_apara_deglist_l2<-asdc_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

#asdc_apara_rank_l2<-asdc_apara_deglist_l2%>%
#  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
#  dplyr::select(GeneSymbol,rankmetric) %>%
#  na.omit() %>%
#  distinct() %>%
#  group_by(GeneSymbol) %>%
#  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
#  arrange(desc(rankmetric)) %>%
#  deframe()

#devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

#gsea_apara_l2_asdc<-NamedGeneRankList2GseaTable(rankedgenes = asdc_apara_rank_l2,
#                                       geneset = "all",
#                                       output_directory = tempdir(),
#                                      filename_prefix = "GSEA",
#                                       sampleSize = 101,
#                                       minSize = 20,
#                                       maxSize = Inf,
#                                       scoreType = "std")

#gsea_apara_l2_asdc<- gsea_apara_l2_asdc%>%
#  as_tibble() %>%
#  arrange(desc(NES)) %>% 
#  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
#  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
#  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
#  arrange(padj)%>%
#filter(padj < 0.20) %>%
#  filter(!grepl("TBA", pathway)) %>%
#  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
#  mutate(neglogpadj = -log10(padj)) %>%
#  mutate(pathway = gsub("gd", "", pathway)) %>%
#  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
#  mutate(pathway = gsub("Vg", "V", pathway)) %>%
#  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
#  mutate(pathway = gsub("_", " ", pathway)) %>%
#  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
#  group_by(module_type) %>%
#  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
#  ungroup() %>%
#  filter(!grepl("TBD", pathway)) %>%
#  arrange(desc(neglogpadj)) %>%
#  droplevels()

##########################################################################################################################################
#b_int
b_int_apara_deglist_l2<-b_int_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_int_apara_rank_l2<-b_int_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_b_int<-NamedGeneRankList2GseaTable(rankedgenes = b_int_apara_rank_l2,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_apara_l2_b_int<- gsea_apara_l2_b_int%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#b_mem
b_mem_apara_deglist_l2<-b_mem_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_mem_apara_rank_l2<-b_mem_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_b_mem<-NamedGeneRankList2GseaTable(rankedgenes = b_mem_apara_rank_l2,
                                           geneset = "all",
                                           output_directory = tempdir(),
                                           filename_prefix = "GSEA",
                                           sampleSize = 101,
                                           minSize = 20,
                                           maxSize = Inf,
                                           scoreType = "std")

gsea_apara_l2_b_mem<- gsea_apara_l2_b_mem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#b_naiv
b_naiv_apara_deglist_l2<-b_naiv_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

b_naiv_apara_rank_l2<-b_naiv_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_b_naiv<-NamedGeneRankList2GseaTable(rankedgenes = b_naiv_apara_rank_l2,
                                            geneset = "all",
                                            output_directory = tempdir(),
                                            filename_prefix = "GSEA",
                                            sampleSize = 101,
                                            minSize = 20,
                                            maxSize = Inf,
                                            scoreType = "std")

gsea_apara_l2_b_naiv<- gsea_apara_l2_b_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#pb
pb_apara_deglist_l2<-pb_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

pb_apara_rank_l2<-pb_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_pb<-NamedGeneRankList2GseaTable(rankedgenes = pb_apara_rank_l2,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_apara_l2_pb<- gsea_apara_l2_pb%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd14_mono
cd14_mono_apara_deglist_l2<-cd14_mono_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd14_mono_apara_rank_l2<-cd14_mono_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd14_mono<-NamedGeneRankList2GseaTable(rankedgenes = cd14_mono_apara_rank_l2,
                                               geneset = "all",
                                               output_directory = tempdir(),
                                               filename_prefix = "GSEA",
                                               sampleSize = 101,
                                               minSize = 20,
                                               maxSize = Inf,
                                               scoreType = "std")

gsea_apara_l2_cd14_mono<- gsea_apara_l2_cd14_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd16_mono
cd16_mono_apara_deglist_l2<-cd16_mono_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd16_mono_apara_rank_l2<-cd16_mono_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd16_mono<-NamedGeneRankList2GseaTable(rankedgenes = cd16_mono_apara_rank_l2,
                                               geneset = "all",
                                               output_directory = tempdir(),
                                               filename_prefix = "GSEA",
                                               sampleSize = 101,
                                               minSize = 20,
                                               maxSize = Inf,
                                               scoreType = "std")

gsea_apara_l2_cd16_mono<- gsea_apara_l2_cd16_mono%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_ctl
cd4_ctl_apara_deglist_l2<-cd4_ctl_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_ctl_apara_rank_l2<-cd4_ctl_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd4_ctl<-NamedGeneRankList2GseaTable(rankedgenes = cd4_ctl_apara_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_apara_l2_cd4_ctl<- gsea_apara_l2_cd4_ctl%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_naiv
cd4_naiv_apara_deglist_l2<-cd4_naiv_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_naiv_apara_rank_l2<-cd4_naiv_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd4_naiv<-NamedGeneRankList2GseaTable(rankedgenes = cd4_naiv_apara_rank_l2,
                                              geneset = "all",
                                              output_directory = tempdir(),
                                              filename_prefix = "GSEA",
                                              sampleSize = 101,
                                              minSize = 20,
                                              maxSize = Inf,
                                              scoreType = "std")

gsea_apara_l2_cd4_naiv<- gsea_apara_l2_cd4_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_prolif
cd4_prolif_apara_deglist_l2<-cd4_prolif_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_prolif_apara_rank_l2<-cd4_prolif_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd4_prolif<-NamedGeneRankList2GseaTable(rankedgenes = cd4_prolif_apara_rank_l2,
                                                geneset = "all",
                                                output_directory = tempdir(),
                                                filename_prefix = "GSEA",
                                                sampleSize = 101,
                                                minSize = 20,
                                                maxSize = Inf,
                                                scoreType = "std")

gsea_apara_l2_cd4_prolif<- gsea_apara_l2_cd4_prolif%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_tcm
cd4_tcm_apara_deglist_l2<-cd4_tcm_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_tcm_apara_rank_l2<-cd4_tcm_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd4_tcm<-NamedGeneRankList2GseaTable(rankedgenes = cd4_tcm_apara_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_apara_l2_cd4_tcm<- gsea_apara_l2_cd4_tcm%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd4_tem
cd4_tem_apara_deglist_l2<-cd4_tem_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd4_tem_apara_rank_l2<-cd4_tem_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd4_tem<-NamedGeneRankList2GseaTable(rankedgenes = cd4_tem_apara_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_apara_l2_cd4_tem<- gsea_apara_l2_cd4_tem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_naiv
cd8_naiv_apara_deglist_l2<-cd8_naiv_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_naiv_apara_rank_l2<-cd8_naiv_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd8_naiv<-NamedGeneRankList2GseaTable(rankedgenes = cd8_naiv_apara_rank_l2,
                                              geneset = "all",
                                              output_directory = tempdir(),
                                              filename_prefix = "GSEA",
                                              sampleSize = 101,
                                              minSize = 20,
                                              maxSize = Inf,
                                              scoreType = "std")

gsea_apara_l2_cd8_naiv<- gsea_apara_l2_cd8_naiv%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_tcm
cd8_tcm_apara_deglist_l2<-cd8_tcm_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_tcm_apara_rank_l2<-cd8_tcm_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd8_tcm<-NamedGeneRankList2GseaTable(rankedgenes = cd8_tcm_apara_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_apara_l2_cd8_tcm<- gsea_apara_l2_cd8_tcm%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cd8_tem
cd8_tem_apara_deglist_l2<-cd8_tem_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cd8_tem_apara_rank_l2<-cd8_tem_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cd8_tem<-NamedGeneRankList2GseaTable(rankedgenes = cd8_tem_apara_rank_l2,
                                             geneset = "all",
                                             output_directory = tempdir(),
                                             filename_prefix = "GSEA",
                                             sampleSize = 101,
                                             minSize = 20,
                                             maxSize = Inf,
                                             scoreType = "std")

gsea_apara_l2_cd8_tem<- gsea_apara_l2_cd8_tem%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#treg
treg_apara_deglist_l2<-treg_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

treg_apara_rank_l2<-treg_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_treg<-NamedGeneRankList2GseaTable(rankedgenes = treg_apara_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_apara_l2_treg<- gsea_apara_l2_treg%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#gdt
gdt_apara_deglist_l2<-gdt_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

gdt_apara_rank_l2<-gdt_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_gdt<-NamedGeneRankList2GseaTable(rankedgenes = gdt_apara_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_apara_l2_gdt<- gsea_apara_l2_gdt%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  #mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#mait
mait_apara_deglist_l2<-mait_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

mait_apara_rank_l2<-mait_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_mait<-NamedGeneRankList2GseaTable(rankedgenes = mait_apara_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_apara_l2_mait<- gsea_apara_l2_mait%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#ilc
ilc_apara_deglist_l2<-ilc_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

ilc_apara_rank_l2<-ilc_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_ilc<-NamedGeneRankList2GseaTable(rankedgenes = ilc_apara_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_apara_l2_ilc<- gsea_apara_l2_ilc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#pdc
pdc_apara_deglist_l2<-pdc_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

pdc_apara_rank_l2<-pdc_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_pdc<-NamedGeneRankList2GseaTable(rankedgenes = pdc_apara_rank_l2,
                                         geneset = "all",
                                         output_directory = tempdir(),
                                         filename_prefix = "GSEA",
                                         sampleSize = 101,
                                         minSize = 20,
                                         maxSize = Inf,
                                         scoreType = "std")

gsea_apara_l2_pdc<- gsea_apara_l2_pdc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#cdc2
cdc2_apara_deglist_l2<-cdc2_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

cdc2_apara_rank_l2<-cdc2_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_cdc2<-NamedGeneRankList2GseaTable(rankedgenes = cdc2_apara_rank_l2,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_apara_l2_cdc2<- gsea_apara_l2_cdc2%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################

##########################################################################################################################################
#nk
nk_apara_deglist_l2<-nk_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nk_apara_rank_l2<-nk_apara_deglist_l2%>%
  mutate(rankmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankmetric = mean(rankmetric)) %>%
  dplyr::filter_all(all_vars(is.finite(.)))%>%
  arrange(desc(rankmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_nk<-NamedGeneRankList2GseaTable(rankedgenes = nk_apara_rank_l2,
                                        geneset = "all",
                                        output_directory = tempdir(),
                                        filename_prefix = "GSEA",
                                        sampleSize = 101,
                                        minSize = 20,
                                        maxSize = Inf,
                                        scoreType = "std")

gsea_apara_l2_nk<- gsea_apara_l2_nk%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>% 
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#nk_cd56br
nk_cd56br_apara_deglist_l2<-nk_cd56br_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nk_cd56br_apara_rank_l2_cd56br<-nk_cd56br_apara_deglist_l2%>%
  mutate(rank_cd56brmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rank_cd56brmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rank_cd56brmetric = mean(rank_cd56brmetric)) %>%
  arrange(desc(rank_cd56brmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_nk_cd56br<-NamedGeneRankList2GseaTable(rankedgenes = nk_cd56br_apara_rank_l2_cd56br,
                                               geneset = "all",
                                               output_directory = tempdir(),
                                               filename_prefix = "GSEA",
                                               sampleSize = 101,
                                               minSize = 20,
                                               maxSize = Inf,
                                               scoreType = "std")

gsea_apara_l2_nk_cd56br<- gsea_apara_l2_nk_cd56br%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

##########################################################################################################################################
#nkprolif
nkprolif_apara_deglist_l2<-nkprolif_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

nkprolif_apara_rank_l2prolif<-nkprolif_apara_deglist_l2%>%
  mutate(rankprolifmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rankprolifmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rankprolifmetric = mean(rankprolifmetric)) %>%
  arrange(desc(rankprolifmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_nkprolif<-NamedGeneRankList2GseaTable(rankedgenes = nkprolif_apara_rank_l2prolif,
                                              geneset = "all",
                                              output_directory = tempdir(),
                                              filename_prefix = "GSEA",
                                              sampleSize = 101,
                                              minSize = 20,
                                              maxSize = Inf,
                                              scoreType = "std")

gsea_apara_l2_nkprolif<- gsea_apara_l2_nkprolif%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  # mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()

#Untitled 4
##########################################################################################################################################
#hspc
hspc_apara_deglist_l2<-hspc_apara_deglist_l2%>%rownames_to_column(., "GeneSymbol")

hspc_rahspc<-hspc_apara_deglist_l2%>%
  mutate(rahspcmetric = -log10(.$p_val)*sign(.$avg_log2FC)) %>%
  dplyr::select(GeneSymbol,rahspcmetric) %>%
  na.omit() %>%
  distinct() %>%
  group_by(GeneSymbol) %>%
  dplyr::summarize(rahspcmetric = mean(rahspcmetric)) %>%
  arrange(desc(rahspcmetric)) %>%
  deframe()

##devtools::source_url("https://github.com/TranLab/ModuleLists/blob/main/NamedGeneRankList2GseaTable.R?raw=TRUE")

gsea_apara_l2_hspc<-NamedGeneRankList2GseaTable(rankedgenes = hspc_rahspc,
                                          geneset = "all",
                                          output_directory = tempdir(),
                                          filename_prefix = "GSEA",
                                          sampleSize = 101,
                                          minSize = 20,
                                          maxSize = Inf,
                                          scoreType = "std")

gsea_apara_l2_hspc<- gsea_apara_l2_hspc%>%
  as_tibble() %>%
  arrange(desc(NES)) %>% 
  dplyr::select(module_type, pathway, ES, NES, size, leadingEdge, pval, padj) %>%
  mutate(leadingEdge = gsub("^c\\(|\\)$", "", leadingEdge)) %>%
  mutate(leadingEdge = gsub('"', "", leadingEdge)) %>%
  arrange(padj)%>%
  #filter(padj < 0.20) %>%
  filter(!grepl("TBA", pathway)) %>%
  dplyr::select( module_type, pathway, leadingEdge, size, NES, padj) %>%
  mutate(neglogpadj = -log10(padj)) %>%
  mutate(pathway = gsub("gd", "", pathway)) %>%
  mutate(pathway = gsub("Vd2", "V2", pathway)) %>%
  mutate(pathway = gsub("Vg", "V", pathway)) %>%
  mutate(pathway = gsub("HALLMARK_", "", pathway)) %>%
  mutate(pathway = gsub("_", " ", pathway)) %>%
  mutate(pathway = sub(".*?\\_", "", pathway)) %>%
  group_by(module_type) %>%
  # mutate(pathway = fct_reorder(pathway, NES, .desc = TRUE)) %>%
  ungroup() %>%
  filter(!grepl("TBD", pathway)) %>%
  arrange(desc(neglogpadj)) %>%
  droplevels()
```

# Concatenate and save GSEA results of each celltype in predictedcelltype L2
```{r concatenate and save GSEA tables for each celltype L1}
# concatenate gsea results
predictedcelltypel2_gsea<- mget(ls(pattern = "gsea_l2*"))

# Save
saveRDS(predictedcelltypel2_gsea, "/Users/rholla/Desktop/R21 Sterile Immunity data and scripts/All scRNA-seq GEX consolidated (CITE-exvivo and multiome)/In vitro/R21SterileImmunity_invitro_CITE_Multiome_allcelltypesL2_GSEA_irbc_vs_urbc_MAST_no_randome_effect_05_01_2023_TranLab_NamedGeneRankList2GseaTable_02_21_2023.rds")
```

#based on only monocytes,lets look at some bubble plots
```{r}
signif_pathways_celltpyel2_list<-imap(all_mono_gsea, ~.x %>%
                                         filter(., neglogpadj>0.698970004)%>%
                                  filter(., !grepl("MSigDB_C7_all_v7.4|MSigDB_C8_all_v7.4|lowBTMs|highBTMs|BloodGen3Module|MonacoModules|MSigDB_C5_GO_bp_v7.4", module_type)))

signif_pathways_celltpyel2<-bind_rows(signif_pathways_celltpyel2_list, .id = "status")

signif_pathways_celltpyel2<-signif_pathways_celltpyel2 %>%
    mutate_at("status",str_replace,"gsea_l2_", "") %>%
    mutate_at("status", toupper)


# Plot as bubble plot
basetextsize <- 8
myfont <- "sans"
bubble_max_size <- 6
# The comparison is Apara vs para, so we're plotting -NES

signif_plot_pred.celltpyel2<-ggplot(signif_pathways_celltpyel2, aes(x = pathway, y=status)) +
                  geom_point(aes(size=neglogpadj, color=NES), alpha = 0.65) +
                  scale_size_area(name = expression(-log[10]~adj.~p~value), max_size = bubble_max_size) +
                  scale_color_gradient2(low = "blue",
                                        mid = "white",
                                        high = "red",
                                        name = "Pf-infected RBC vs uninfected RBC 20% cut off") +
                  hrbrthemes::theme_ipsum_es(base_family = myfont, base_size = basetextsize) +
                  theme(axis.title.x = element_blank(),
                        axis.title.y = element_blank(),
                        strip.background = element_blank(),
                        legend.position = "bottom",
                        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+
  facet_wrap(~module_type, scales = "free")

pdf("~/gsea_mono_invitro.pdf",height=15, width=12)
```

